{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create ISA objects for EATRIS-Plus multi-omics data set of Czech population cohort\n",
    "\n",
    "See example: https://isatools.readthedocs.io/en/latest/example-createSimpleISAtab.html"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Define ontology references\n",
    "\n",
    "Onotologies in (FAIR genomes metadata schema)[https://github.com/fairgenomes/fairgenomes-semantic-model] include AFR, AFRL, DC (Dublin Core), DUO (Data Use Ontology), EDAM, EFO, FG, FIX, GAZ, GENEPIO, GO, GSSO, HANCESTRO, HGNC, HL7, HP, IAO (Information Artifact Ontology), ICO (Informed Consent Ontology), LOINC, NCIT, OBI, OMIABIS, Orphanet, PATO, RO (Relation Ontology), SIO (Semanticscience Integrated Ontology), SNOMEDCT, SWO, UATC.\n",
    "\n",
    "We reuse these as much as possible."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "from isatools.model import *\n",
    "ontologies = {\n",
    "    \"BAO\": OntologySource(\n",
    "        name = \"BAO - BioAssay Ontology\", \n",
    "        file = \"http://www.bioassayontology.org/bao/bao_complete.owl\",\n",
    "        description = \"The BioAssay Ontology (BAO) describes biological screening assays and their results including high-throughput screening (HTS) data for the purpose of categorizing assays and data analysis. BAO is an extensible, knowledge-based, highly expressive (currently SHOIQ(D)) description of biological assays making use of descriptive logic based features of the Web Ontology Language (OWL). BAO currently has over 700 classes and also makes use of several other ontologies. It describes several concepts related to biological screening, including Perturbagen, Format, Meta Target, Design, Detection Technology, and Endpoint. Perturbagens are perturbing agents that are screened in an assay; they are mostly small molecules. Assay Meta Target describes what is known about the biological system and / or its components interrogated in the assay (and influenced by the Perturbagen). Meta target can be directly described as a molecular entity (e.g. a purified protein or a protein complex), or indirectly by a biological process or event (e.g. phosphorylation). Format describes the biological or chemical features common to each test condition in the assay and includes biochemical, cell-based, organism-based, and variations thereof. The assay Design describes the assay methodology and implementation of how the perturbation of the biological system is translated into a detectable signal. Detection Technology relates to the physical method and technical details to detect and record a signal. Endpoints are the final HTS results as they are usually published (such as IC50, percent inhibition, etc). BAO has been designed to accommodate multiplexed assays. All main BAO components include multiple levels of sub-categories and specification classes, which are linked via object property relationships forming an expressive knowledge-based representation.\"), \n",
    "    \"CHEBI\": OntologySource(\n",
    "        name = \"CHEBI - Chemical Entities of Biological Interest\", \n",
    "        file = \"http://purl.obolibrary.org/obo/chebi.owl\",\n",
    "        description = \"A structured classification of molecular entities of biological interest focusing on 'small' chemical compounds.\"), \n",
    "    \"CHMO\": OntologySource(\n",
    "        name = \"CHMO - Chemical Methods Ontology\", \n",
    "        file = \"http://purl.obolibrary.org/obo/chmo.owl\",\n",
    "        description = \"CHMO, the chemical methods ontology, describes methods used to collect data in chemical experiments, such as mass spectrometry and electron microscopy prepare and separate material for further analysis, such as sample ionisation, chromatography, and electrophoresis synthesise materials, such as epitaxy and continuous vapour deposition It also describes the instruments used in these experiments, such as mass spectrometers and chromatography columns. It is intended to be complementary to the Ontology for Biomedical Investigations (OBI).\"), \n",
    "    \"CRO\": OntologySource(\n",
    "        # The Contributor Role Ontology (CRO) is an extension of the CASRAI Contributor Roles Taxonomy (CRediT) and replaces the former Contribution Ontology.\n",
    "        name = \"CRO - Contributor Role Ontology\",\n",
    "        file = \"http://purl.obolibrary.org/obo/cro.owl\",\n",
    "        description = \"A classification of the diverse roles performed in the work leading to a published research output in the sciences. Its purpose to provide transparency in contributions to scholarly published work, to enable improved systems of attribution, credit, and accountability.\"),\n",
    "    \"EDAM\": OntologySource(\n",
    "        name = \"EDAM - EMBRACE Data and Methods\", \n",
    "        file = \"http://edamontology.org/EDAM.owl\",\n",
    "        description = \"EDAM (EMBRACE Data and Methods) is an ontology of common bioinformatics operations, topics, types of data including identifiers, and formats. EDAM comprises common concepts (shared within the bioinformatics community) that apply to semantic annotation of resources.\"), \n",
    "    \"EFO\": OntologySource(\n",
    "        name = \"EFO - Experimental Factor Ontology\", \n",
    "        file = \"http://www.ebi.ac.uk/efo/efo.owl\",\n",
    "        description = \"The Experimental Factor Ontology (EFO) provides a systematic description of many experimental variables available in EBI databases, and for external projects such as the NHGRI GWAS catalogue. It combines parts of several biological ontologies, such as anatomy, disease and chemical compounds. The scope of EFO is to support the annotation, analysis and visualization of data handled by many groups at the EBI and as the core ontology for OpenTargets.org\"), \n",
    "    \"GENEPIO\": OntologySource(\n",
    "        name = \"GENEPIO - Genomic Epidemiology Ontology\", \n",
    "        file = \"http://purl.obolibrary.org/obo/genepio.owl\",\n",
    "        description = \"The Genomic Epidemiology Ontology (GenEpiO) covers vocabulary necessary to identify, document and research foodborne pathogens and associated outbreaks.\"),\n",
    "    \"MI\": OntologySource(\n",
    "        name = \"MI - Molecular Interactions Controlled Vocabulary\", \n",
    "        file = \"http://purl.obolibrary.org/obo/mi.owl\",\n",
    "        description = \"A structured controlled vocabulary for the annotation of experiments concerned with protein-protein interactions.\"),\n",
    "    \"MMO\": OntologySource(\n",
    "        name = \"MMO - Measurement method ontology\", \n",
    "        file = \"http://purl.obolibrary.org/obo/mmo.owl\",\n",
    "        description = \"A representation of the variety of methods used to make clinical and phenotype measurements.\"),\n",
    "    \"MS\": OntologySource(\n",
    "        name = \"Metabolomics Standards Initiative Ontology (MSIO)\",\n",
    "        file = \"http://purl.obolibrary.org/obo/ms.owl\",\n",
    "        description = \"A structured controlled vocabulary for the annotation of experiments concerned with proteomics mass spectrometry.\"),\n",
    "    \"MSIO\": OntologySource(\n",
    "        name = \"MS - Mass spectrometry ontology\",\n",
    "        file = \"http://purl.obolibrary.org/obo/msio.owl\",\n",
    "        description = \"MSIO aims to provide a single point of entry to support semantic markup of experiments making use of NMR and MS techniques to identify, measure and quantify small molecules known as metabolites. MSIO covers metabolite profiling, targeted or undertargeted, tracer based applications. MSIO reuses a number of resources such as CHEBI, DUO, NMRCV, OBI, and STATO.\"),\n",
    "    \"NCBITAXON\": OntologySource(\n",
    "        name = \"NCBI organismal classification\", \n",
    "        file = \"http://purl.obolibrary.org/obo/ncbitaxon.owl\",\n",
    "        description = \"An ontology representation of the NCBI organismal taxonomy\"),\n",
    "    \"NCIT\": OntologySource(\n",
    "        name = \"NCI Thesaurus OBO Edition\", \n",
    "        file = \"http://purl.obolibrary.org/obo/ncit.owl\",\n",
    "        description = \"The NCIt OBO Edition project aims to increase integration of the NCIt with OBO Library ontologies. NCIt is a reference terminology that includes broad coverage of the cancer domain, including cancer related diseases, findings and abnormalities. NCIt OBO Edition releases should be considered experimental.\"),\n",
    "    \"OBI\": OntologySource(\n",
    "        name = \"OBI - Ontology for Biomedical Investigations\", \n",
    "        file = \"http://purl.obolibrary.org/obo/obi.owl\",\n",
    "        description = \"An integrated ontology for the description of life-science and clinical investigations\"),\n",
    "    \"OMIABIS\": OntologySource(\n",
    "        name = \"Ontologized MIABIS\", \n",
    "        file = \"http://purl.obolibrary.org/obo/omiabis.owl\",\n",
    "        description = \"An ontological version of MIABIS (Minimum Information About BIobank data Sharing)\"),\n",
    "    \"ORNASEQ\": OntologySource(\n",
    "        name = \"Ontology for RNA sequencing (ORNASEQ)\", \n",
    "        file = \"http://purl.obolibrary.org/obo/ornaseq.owl\",\n",
    "        description = \"An application ontology designed to annotate next-generation sequencing experiments performed on RNA.\"),\n",
    "    \"PRIDE\": OntologySource(\n",
    "        name = \"PRIDE Controlled Vocabulary\",\n",
    "        file = \"http://purl.obolibrary.org/obo/pride_cv.obo\",\n",
    "        description = \"The PRIDE PRoteomics IDEntifications (PRIDE) database is a centralized, standards compliant, public data repository for proteomics data, including protein and peptide identifications, post-translational modifications and supporting spectral evidence.\"),\n",
    "    \"STATO\": OntologySource(\n",
    "        name = \"STATO: the statistical methods ontology\",\n",
    "        file = \"http://purl.obolibrary.org/obo/stato.owl\",\n",
    "        description = \"STATO is the statistical methods ontology. It contains concepts and properties related to statistical methods, probability distributions and other concepts related to statistical analysis, including relationships to study designs and plots.\"),\n",
    "    \"UBERON\": OntologySource(\n",
    "        name = \"Uber-anatomy ontology\",\n",
    "        file = \"http://purl.obolibrary.org/obo/uberon.owl\",\n",
    "        description = \"Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.\"),\n",
    "    \"tbd\": OntologySource(\n",
    "        name = \"to be defined\",\n",
    "        file = \"http://tbd.owl\",\n",
    "        description = \"tbd\")\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create investigation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "investigation_filename = \"i_investigation.txt\"\n",
    "investigation = Investigation(\n",
    "    filename = investigation_filename, \n",
    "    identifier = \"\", \n",
    "    title = \"EATRIS-Plus - Flagship in Personalised Medicine\",\n",
    "    description = \"EATRIS-Plus project aims to support the long-term sustainability of the European Research Infrastructure for Translational Medicine (EATRIS) by delivering innovative scientific tools to the research community, strengthening the EATRIS financial model, and reinforcing EATRIS' leadership in the European Research Area in the field of Personalised Medicine (PM).\",\n",
    "    submission_date = \"\",\n",
    "    public_release_date = \"\",\n",
    "    ontology_source_references = [o for o in ontologies.values()],\n",
    "    publications = None,\n",
    "    contacts = [\n",
    "        Person(\n",
    "            last_name = \"Keidong\", \n",
    "            first_name = \"Eliis\",\n",
    "            #mid_initials = \"\",\n",
    "            affiliation = \"EATRIS\",\n",
    "            roles = [\n",
    "                OntologyAnnotation(\n",
    "                    term = \"project management role\",\n",
    "                    term_source = ontologies[\"CRO\"], \n",
    "                    term_accession =\"http://purl.obolibrary.org/obo/CRO_0000065\")])],\n",
    "    studies = None,\n",
    "    comments = None)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create study"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "study_filename = \"s_study.txt\"\n",
    "cohort_study = Study(\n",
    "    filename = study_filename, \n",
    "    identifier = \"\", \n",
    "    title = \"Multi-omics data of a Czech population cohort\",\n",
    "    description = \"Multi-omics data of a Czech population cohort\", \n",
    "    submission_date = \"\", \n",
    "    public_release_date = \"\",\n",
    "    contacts = [\n",
    "        Person(\n",
    "            last_name = \"Hajduch\", \n",
    "            first_name = \"Marian\",\n",
    "            #mid_initials = \"\",\n",
    "            affiliation = \"Institute of Molecular and Translational Medicine (IMTM), Palacky University Olomouc\")],\n",
    "    design_descriptors = [\n",
    "        OntologyAnnotation(\n",
    "                term = \"Multi-omics study\",\n",
    "                term_source = ontologies[\"PRIDE\"],\n",
    "                term_accession = \"http://purl.obolibrary.org/obo/PRIDE_0000461\"),\n",
    "        OntologyAnnotation(\n",
    "                term = \"population based study design\",\n",
    "                term_source = ontologies[\"OMIABIS\"],\n",
    "                term_accession = \"http://purl.obolibrary.org/obo/OMIABIS_0001022\")], \n",
    "    factors = None, \n",
    "    protocols = None,\n",
    "    assays = None,\n",
    "    sources = None,\n",
    "    samples = None,\n",
    "    process_sequence = None,\n",
    "    other_material = None,\n",
    "    characteristic_categories = None,\n",
    "    comments = None,\n",
    "    units = None)\n",
    "investigation.studies.append(cohort_study)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Samples, protocols, assays"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Define protocols and protocol parameters (for all assays)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# protocol parameters \n",
    "protocol_params = {\n",
    "    #\"anatomical entity\": ProtocolParameter(\n",
    "    #    parameter_name = OntologyAnnotation(\n",
    "    #        term = \"anatomical entity\",\n",
    "    #        term_source = ontologies[\"UBERON\"],\n",
    "    #        term_accession = \"http://purl.obolibrary.org/obo/UBERON_0001062\")),\n",
    "    \"Post Extraction\": ProtocolParameter(\n",
    "        parameter_name = \"Post Extraction\"),\n",
    "    \"Derivatization\": ProtocolParameter(\n",
    "        parameter_name = OntologyAnnotation(\n",
    "            term = \"Derivatization\",\n",
    "            term_source = ontologies[\"MSIO\"],\n",
    "            term_accession = \"http://purl.obolibrary.org/obo/MSIO_0000111\")),\n",
    "    \"Chromatography Instrument\": ProtocolParameter(\n",
    "        parameter_name = OntologyAnnotation(\n",
    "            term = \"Chromatography Instrument\",\n",
    "            term_source = ontologies[\"OBI\"],\n",
    "            term_accession = \"http://purl.obolibrary.org/obo/OBI_0000485\")),\n",
    "    \"Column model\": ProtocolParameter(\n",
    "        parameter_name = \"Column model\"),\n",
    "    \"Column type\": ProtocolParameter(\n",
    "        parameter_name = \"Column type\"),\n",
    "    \"Scan polarity\": ProtocolParameter(\n",
    "        parameter_name = OntologyAnnotation(\n",
    "            term = \"scan polarity\",\n",
    "            term_source = ontologies[\"MS\"],\n",
    "            term_accession = \"http://purl.obolibrary.org/obo/MS_1000465\")),\n",
    "        #values: \n",
    "        #negative scan http://purl.obolibrary.org/obo/MS_1000129\n",
    "        #positive scan http://purl.obolibrary.org/obo/MS_1000130\n",
    "    \"Scan m/z range\": ProtocolParameter(\n",
    "        parameter_name = OntologyAnnotation(\n",
    "            term = \"Scan m/z range\")),\n",
    "            #term_source = ontologies[\"\"],\n",
    "            #term_accession = \"\")),\n",
    "    \"Instrument\": ProtocolParameter(\n",
    "        parameter_name = OntologyAnnotation(\n",
    "            term = \"Instrument\",\n",
    "            term_source = ontologies[\"MS\"],\n",
    "            term_accession = \"http://purl.obolibrary.org/obo/MS_1000463\")),\n",
    "    \"Ion source\": ProtocolParameter(\n",
    "        parameter_name = OntologyAnnotation(\n",
    "            term = \"Ion source\",\n",
    "            term_source = ontologies[\"CHMO\"],\n",
    "            term_accession = \"http://purl.obolibrary.org/obo/CHMO_0000960\")),\n",
    "    \"Mass analyzer\": ProtocolParameter(\n",
    "        parameter_name = OntologyAnnotation(\n",
    "            term = \"Mass analyzer\",\n",
    "            term_source = ontologies[\"MS\"],\n",
    "            term_accession = \"http://purl.obolibrary.org/obo/MS_1000451\")),\n",
    "    \"method reference\": ProtocolParameter(\n",
    "        parameter_name = OntologyAnnotation(\n",
    "            term = \"method reference\",\n",
    "            term_source = ontologies[\"MI\"],\n",
    "            term_accession = \"http://purl.obolibrary.org/obo/MI_0357\")),\n",
    "    \"technical replicate\": ProtocolParameter(\n",
    "        parameter_name = OntologyAnnotation(\n",
    "            term = \"technical replicate\",\n",
    "            term_source = ontologies[\"EFO\"],\n",
    "            term_accession = \"http://www.ebi.ac.uk/efo/EFO_0002090\"))\n",
    "}\n",
    "\n",
    "# protocols\n",
    "protocols = {\n",
    "    \"sample_collection\": Protocol( \n",
    "        name = \"sample_collection_protocol\",\n",
    "        protocol_type = OntologyAnnotation(\n",
    "            term = \"sample collection\", # has to be \"sample collection\" based on ISA specification, but no term available\n",
    "            term_source = None,  \n",
    "            term_accession = None)),\n",
    "        #parameters = [\n",
    "        #    protocol_params[\"anatomical entity\"]\n",
    "            # additional parameters could include, e.g. collection or storage procedure]\n",
    "    \"dna_extraction\": Protocol(\n",
    "        name = \"DNA extraction\",\n",
    "        protocol_type = OntologyAnnotation(\n",
    "            term = \"DNA extraction\",\n",
    "            term_source = ontologies[\"OBI\"], \n",
    "            term_accession = \"http://purl.obolibrary.org/obo/OBI_0000257\")),\n",
    "    \"rna_extraction\": Protocol(\n",
    "        name = \"RNA extraction\",\n",
    "        protocol_type = OntologyAnnotation(\n",
    "            term = \"RNA extraction\",\n",
    "            term_source = ontologies[\"OBI\"], \n",
    "            term_accession = \"http://purl.obolibrary.org/obo/OBI_0666666\"),\n",
    "        description = \"RNA extraction is a nucleic acid extraction where the desired output material is RNA\"),\n",
    "    \"Whole Genome Sequencing\": Protocol(\n",
    "        name = \"Whole Genome Sequencing\", \n",
    "        protocol_type = OntologyAnnotation(\n",
    "            term = \"Whole Genome Sequencing\",\n",
    "            term_source = ontologies[\"NCIT\"],\n",
    "            term_accession = \"http://purl.obolibrary.org/obo/NCIT_C101294\")),\n",
    "    \"mRNA library preparation\": Protocol(\n",
    "        name = \"mRNA library preparation\", \n",
    "        protocol_type = OntologyAnnotation(\n",
    "            term = \"library preparation protocol\",\n",
    "            term_source = ontologies[\"ORNASEQ\"],\n",
    "            term_accession = \"http://purl.obolibrary.org/obo/ORNASEQ_0000007\"),\n",
    "        description = \"Library preparation from 800 ng of total RNA was performed according to Illumina stranded mRNA prep Reference Guide (Illumina, San Diego, CA, USA). ). Library quality check was performed using LabChip GX Touch HT High Sensitivity assay (PerkinElmer, USA). The libraries were quantified for sequencing using KAPA Library Quantification Kit (KAPA Biosystems, Wilmington, MA, USA).\",\n",
    "        uri = \"https://www.illumina.com/products/by-type/sequencing-kits/library-prep-kits/stranded-mrna-prep.html\"),\n",
    "    \"mRNA Sequencing\": Protocol(\n",
    "        name = \"mRNA Sequencing\", \n",
    "        protocol_type = OntologyAnnotation(\n",
    "            term = \"nucleic acid sequencing\", #\"mRNA Sequencing\",\n",
    "            term_source = ontologies[\"NCIT\"],\n",
    "            term_accession = \"http://purl.obolibrary.org/obo/NCIT_C18881\"),\n",
    "        description = \"Sequencing was performed with Illumina NovaSeq system using S4 flow cell with lane divider (Illumina, San Diego, CA, USA). Read length for the paired-end run was 2x151 bp.\"),\n",
    "    \"microRNAseq\": Protocol(\n",
    "        name = \"MicroRNA Sequencing\", \n",
    "        protocol_type = OntologyAnnotation(\n",
    "            term = \"MicroRNA Sequencing\",\n",
    "            term_source = ontologies[\"NCIT\"],\n",
    "            term_accession = \"http://purl.obolibrary.org/obo/NCIT_C156057\")),\n",
    "    # microRNA qRT-PCR\n",
    "    \"Reverse Transcription\": Protocol(\n",
    "        name = \"Reverse Transcription\", \n",
    "        protocol_type = OntologyAnnotation(\n",
    "            term = \"artificially induced reverse transcription\",\n",
    "            term_source = ontologies[\"OBI\"],\n",
    "            term_accession = \"http://purl.obolibrary.org/obo/OBI_0600028\"),\n",
    "        description = \"A protocol with the objective to transcribe single-stranded RNA into complementary DNA (cDNA)\"),\n",
    "    \"miRNA array\": Protocol(\n",
    "        name = \"miRNA array\", \n",
    "        protocol_type = OntologyAnnotation(\n",
    "            term = \"microRNA profiling by RT-PCR\",\n",
    "            term_source = ontologies[\"EFO\"],\n",
    "            term_accession = \"http://www.ebi.ac.uk/efo/EFO_0007687\"),\n",
    "        parameters = [protocol_params[\"technical replicate\"]],\n",
    "        description = \"An assay in which a set of microRNAs of a biological sample is analysed by reverse transcription PCR (RT-PCR)\"),\n",
    "    \"Data Normalization\": Protocol(\n",
    "        name = \"Data Normalization\", \n",
    "        protocol_type = OntologyAnnotation(\n",
    "            term = \"Constant Normalization\",\n",
    "            term_source = ontologies[\"NCIT\"],\n",
    "            term_accession = \"http://purl.obolibrary.org/obo/NCIT_C63786\"),\n",
    "        description = \"Comparison of a variable factor to one with a fixed value (e.g., comparison of gene expression to a gene with constant expression)\"),\n",
    "    # metabolomics protocols\n",
    "    \"Extraction\": Protocol(\n",
    "        name = \"Extraction\", \n",
    "        protocol_type = OntologyAnnotation(\n",
    "            term = \"extraction\",\n",
    "            term_source = ontologies[\"OBI\"],\n",
    "            term_accession = \"http://purl.obolibrary.org/obo/OBI_0302884\"), #also in: MSIO, etc.\n",
    "        parameters = [protocol_params[\"Post Extraction\"],\n",
    "                      protocol_params[\"Derivatization\"]]),\n",
    "        #description = \"<i>Sample preparation:</i> To 50 uL plasma or serum were added 50 uL standard 1 (23.5 umol/L <sup>2</sup>H<sub>3</sub>-free carnitine in H<sub>2</sub>O) and 50 uL standard 2 (10 umol/L <sup>2</sup>H<sub>3</sub>-C<sub>2</sub>-, 2 umol/L <sup>2</sup>H<sub>3</sub>-C<sub>8</sub>- and 2 umol/L <sup>2</sup>H<sub>3</sub>-C<sub>16</sub>-carnitine in acetonitrile). Samples were mixed and subsequently deproteinized with 500 uL of acetonitrile and centrifuged. The resulting supernatant was dried under nitrogen at 45°C, and subsequently derivatized in 100 uL butanol-HCl for 15 min at 60°C. Samples were dried under nitrogen at 45°C and redissolved in 300 uL of acetonitrile. Prior to injection, 70 uL of the acetonitrile containing the acylcarnitines was mixed with 30 uL H<sub>2</sub>O.\",\n",
    "        #uri = \"https://doi.org/10.1023/A:1005587617745\"\n",
    "        #parameters = [\n",
    "         # control samples, standards, etc.   \n",
    "        # Post Extraction\n",
    "        # Derivatization]\n",
    "    \"Chromatography\": Protocol(\n",
    "        name = \"Chromatography\", \n",
    "        protocol_type = OntologyAnnotation(\n",
    "            term = \"chromatography\",\n",
    "            term_source = ontologies[\"CHMO\"],\n",
    "            term_accession = \"http://purl.obolibrary.org/obo/CHMO_0001000\"), #also in: PRIDE\n",
    "        parameters = [protocol_params[\"Chromatography Instrument\"],\n",
    "                      protocol_params[\"Column model\"],\n",
    "                      protocol_params[\"Column type\"]]),\n",
    "      # parameters instrument, column, mobile phase, gradient, settings, injection volume\n",
    "        # Chromatography Instrument\n",
    "        # Column type\n",
    "        # Column model\n",
    "        # Guard column\n",
    "        # Autosampler model\n",
    "    \"Mass spectrometry\": Protocol(\n",
    "        name = \"Mass spectrometry\", \n",
    "        protocol_type = OntologyAnnotation(\n",
    "            term = \"mass spectrometry\",\n",
    "            term_source = ontologies[\"CHMO\"],\n",
    "            term_accession = \"http://purl.obolibrary.org/obo/CHMO_0000470\"), #also in: MSIO, PRIDE\n",
    "        #description = \"<i>Sample introduction and ESI-MS/MS analysis:</i> Free carnitine and acylcarnitines were measured using scanning for precursor ions of mass 85 from 200 to 550 Da during 2 min on a Micromass Quattro II triple-quadrupole mass spectrometer, using a Gilson 231XL autosampler and a Hewlett-Packard HP-1100 HPLC pump, essentially as described previously (Rashed et al 1995a, 1997).\",\n",
    "        #uri = \"https://doi.org/10.1023/A:1005587617745\",\n",
    "        parameters = [protocol_params[\"Scan polarity\"],\n",
    "                      protocol_params[\"Scan m/z range\"],\n",
    "                      protocol_params[\"Instrument\"],\n",
    "                      protocol_params[\"Ion source\"],\n",
    "                      protocol_params[\"Mass analyzer\"],\n",
    "                      protocol_params[\"method reference\"]]),\n",
    "        #  instrument used (make & manufacturer), ion source, ionisation mode (positive / negative), m/z range, and specific parameters such as temperatures, voltages, flow rates, scan rates\n",
    "    \"Data transformation_acylcarnitines\": Protocol(\n",
    "        name = \"Data transformation - Plasma acylcarnitine analysis\", \n",
    "        protocol_type = OntologyAnnotation(\n",
    "            term = \"data transformation\",\n",
    "            term_source = ontologies[\"OBI\"],\n",
    "            term_accession = \"http://purl.obolibrary.org/obo/OBI_0200000\"), #also in: EFO, GENEPIO, OMIABIS, STATO, etc.\n",
    "        description = \"<i>Calibration:</i> Calibration curves were obtained for free carnitine in the range 5-100 umol/L, for acetylcarnitine in the range 2-40 umol/L and for all other available acylcarnitines in the range 0.25-6 umol/L by adding standards to a normal plasma pool. All calibration curves were linear (r > 0.99, data not shown). For unsaturated and hydroxylated acylcarnitines an identical response to that of their saturated counterparts was assumed.\",\n",
    "        uri = \"https://doi.org/10.1023/A:1005587617745\"),\n",
    "        # methods / pipelines and software used to transform the raw data\n",
    "    \"Metabolite identification_acylcarnitines\": Protocol(\n",
    "        name = \"Metabolite identification - Plasma acylcarnitine analysis\", \n",
    "        protocol_type = OntologyAnnotation(\n",
    "            term = \"metabolite identification\",\n",
    "            term_source = ontologies[\"MI\"],\n",
    "            term_accession = \"http://purl.obolibrary.org/obo/MI_2131\")),\n",
    "    # details of methods / pipelines, reference databases and software used to identify features and/or annotate metabolites\n",
    "    \"EM-seq\": Protocol(\n",
    "        name = \"EM-seq\", \n",
    "        protocol_type = OntologyAnnotation(\n",
    "            term = \"methylation profiling by high throughput sequencing\",\n",
    "            term_source = ontologies[\"EFO\"],\n",
    "            term_accession = \"http://www.ebi.ac.uk/efo/EFO_0002761\"),\n",
    "        description = \"NEBNext Enzymatic Methyl-seq (EM-seq)\"),\n",
    "}\n",
    "\n",
    "# append to study protocols\n",
    "for protocol in protocols.values():\n",
    "    cohort_study.protocols.append(protocol)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Define sources and derived samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# add dummy samples\n",
    "for source_idx in range(1, 4):\n",
    "    # create source (=individual)\n",
    "    source_name = \"individual_{0}\".format(source_idx)\n",
    "    source = Source(\n",
    "        name = source_name,\n",
    "        characteristics = [\n",
    "            Characteristic(\n",
    "                category = OntologyAnnotation(\n",
    "                    term = \"Organism\",\n",
    "                    term_source = ontologies[\"OBI\"],\n",
    "                    term_accession = \"http://purl.obolibrary.org/obo/OBI_0100026\"),\n",
    "                value = OntologyAnnotation(\n",
    "                    term = \"Homo sapiens\",\n",
    "                    term_source = ontologies[\"NCBITAXON\"],\n",
    "                    term_accession = \"http://purl.obolibrary.org/obo/NCBITaxon_9606\"))])\n",
    "    cohort_study.sources.append(source)\n",
    "    # create sample\n",
    "    sample_name = \"sample_{0}\".format(source_idx)\n",
    "    sample = Sample(\n",
    "        name = sample_name, \n",
    "        derives_from = [source])\n",
    "    sample.characteristics.append(\n",
    "        Characteristic(\n",
    "            category = OntologyAnnotation(\n",
    "                term = \"anatomical entity\",\n",
    "                term_source = ontologies[\"UBERON\"],\n",
    "                term_accession = \"http://purl.obolibrary.org/obo/UBERON_0001062\"),\n",
    "            value = OntologyAnnotation(\n",
    "                term = \"blood\",\n",
    "                term_source = ontologies[\"UBERON\"],\n",
    "                term_accession = \"http://purl.obolibrary.org/obo/UBERON_0000178\")))\n",
    "    cohort_study.samples.append(sample)\n",
    "    # sample collection process\n",
    "    sample_collection_process = Process(\n",
    "        name = \"samplecollection_{0}\".format(source_idx),\n",
    "        executes_protocol = protocols[\"sample_collection\"],\n",
    "        #parameter_values = [\n",
    "        #    ParameterValue(\n",
    "        #        category = protocol_params[\"anatomical entity\"], #ProtocolParameter \n",
    "        #        value = OntologyAnnotation(\n",
    "        #            term = \"blood\",\n",
    "        #            term_source = ontologies[\"UBERON\"],\n",
    "        #            term_accession = \"http://purl.obolibrary.org/obo/UBERON_0000178\"))],\n",
    "        inputs = [source],\n",
    "        outputs = [sample])\n",
    "    cohort_study.process_sequence.append(sample_collection_process)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Define assays\n",
    "\n",
    "See assay options: https://github.com/ISA-tools/isa-api/blob/master/isatools/resources/config/yaml/assay-options.yml"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "assays = {\n",
    "    \"genomics_imtm\": Assay(\n",
    "        # options from https://github.com/ISA-tools/isa-api/blob/master/isatools/resources/config/yaml/assay-options.yml\n",
    "        # measurement type: genome sequencing\n",
    "        # technology type: nucleic acid sequencing\n",
    "        # protocol type: nucleic acid sequencing\n",
    "        # raw data file: Raw Data File \n",
    "        # derived data file: Derived Data File\n",
    "        filename = \"a_assay_genomics_imtm.txt\",\n",
    "        measurement_type = OntologyAnnotation(\n",
    "            term = \"Whole Genome Sequencing\", #\"DNA Sequence\",\n",
    "            term_source = ontologies[\"NCIT\"],\n",
    "            term_accession = \"http://purl.obolibrary.org/obo/NCIT_C101294\"), #\"http://purl.obolibrary.org/obo/NCIT_C13299\"),\n",
    "        technology_type = OntologyAnnotation(\n",
    "            term = \"Nucleic Acid Sequencing\", #\"Whole Genome Sequencing\",\n",
    "            term_source = ontologies[\"NCIT\"], \n",
    "            term_accession = \"http://purl.obolibrary.org/obo/NCIT_C18881\"), #\"http://purl.obolibrary.org/obo/NCIT_C101294\"),\n",
    "        technology_platform = OntologyAnnotation(\n",
    "            term = \"Illumina platform\", # TODO is there a more specific term for the platform?\n",
    "            term_source = ontologies[\"GENEPIO\"], \n",
    "            term_accession = \"http://purl.obolibrary.org/obo/GENEPIO_0001923\")\n",
    "    ), \n",
    "    \"methylseq_uu\": Assay(\n",
    "        # measurement type: DNA methylation profiling\n",
    "        # technology type: nucleic acid sequencing\n",
    "        # protocol type: nucleic acid sequencing\n",
    "        # raw data file: Raw Data File \n",
    "        # derived data file: Derived Data File\n",
    "        filename = \"a_assay_methylseq_uu.txt\",\n",
    "        measurement_type = OntologyAnnotation(\n",
    "            term = \"DNA methylation profiling by high throughput sequencing assay\", #\"methylation profiling by high throughput sequencing\",\n",
    "            term_source = ontologies[\"OBI\"], #\"EFO\"],\n",
    "            term_accession = \"http://purl.obolibrary.org/obo/OBI_0001266\"), #\"http://www.ebi.ac.uk/efo/EFO_0002761\")\n",
    "        technology_type = OntologyAnnotation(\n",
    "            term = \"Nucleic Acid Sequencing\", \n",
    "            term_source = ontologies[\"NCIT\"], \n",
    "            term_accession = \"http://purl.obolibrary.org/obo/NCIT_C18881\"), \n",
    "        #technology_type = \"Enzymatic methylation sequencing\",\n",
    "        technology_platform = \"NEBNext Enzymatic Methyl-seq (EM-seq)\" # TODO check if this should be the sequencing platform instead\n",
    "    ), \n",
    "    \"rnaseq_fimm\": Assay(\n",
    "        # measurement type: transcription profiling\n",
    "        # technology type: nucleic acid sequencing\n",
    "        # protocol type: nucleic acid sequencing\n",
    "        # raw data file: Raw Data File \n",
    "        # derived data file: Derived Data File\n",
    "        filename = \"a_assay_rnaseq_fimm.txt\",\n",
    "        measurement_type = OntologyAnnotation(\n",
    "            term = \"transcription profiling\", #\"cDNA Sequence\"\n",
    "            term_source = ontologies[\"EFO\"],\n",
    "            term_accession = \"http://www.ebi.ac.uk/efo/EFO_0001032\"),\n",
    "        technology_type = OntologyAnnotation(\n",
    "            term = \"Nucleic Acid Sequencing\", #\"mRNA Sequencing\",\n",
    "            term_source = ontologies[\"NCIT\"], \n",
    "            term_accession = \"http://purl.obolibrary.org/obo/NCIT_C18881\"),\n",
    "        technology_platform = OntologyAnnotation(\n",
    "            term = \"Illumina NovaSeq 6000\",\n",
    "            term_source = ontologies[\"OBI\"], \n",
    "            term_accession = \"http://purl.obolibrary.org/obo/OBI_0002630\")\n",
    "    ),\n",
    "    \"mirnaseq_fimm\": Assay(\n",
    "        # measurement type: transcription profiling\n",
    "        # technology type: nucleic acid sequencing\n",
    "        # protocol type: nucleic acid sequencing\n",
    "        # raw data file: Raw Data File \n",
    "        # derived data file: Derived Data File\n",
    "        filename = \"a_assay_mirnaseq_fimm.txt\",\n",
    "        measurement_type = OntologyAnnotation(\n",
    "            term = \"transcription profiling\", #\"cDNA Sequence\"\n",
    "            term_source = ontologies[\"EFO\"],\n",
    "            term_accession = \"http://www.ebi.ac.uk/efo/EFO_0001032\"),\n",
    "        technology_type = OntologyAnnotation(\n",
    "            term = \"Nucleic Acid Sequencing\", #\"microRNA Sequencing\",\n",
    "            term_source = ontologies[\"NCIT\"], \n",
    "            term_accession = \"http://purl.obolibrary.org/obo/NCIT_C18881\"),\n",
    "        technology_platform = OntologyAnnotation(\n",
    "            term = \"Illumina NovaSeq 6000\",\n",
    "            term_source = ontologies[\"OBI\"], \n",
    "            term_accession = \"http://purl.obolibrary.org/obo/OBI_0002630\")\n",
    "    ), \n",
    "    \"mirnaqrtpcr_sermas\": Assay(\n",
    "        # measurement type: transcription profiling\n",
    "        # technology type: RT-pcr\n",
    "        # protocol type: RT-pcr\n",
    "        # raw data file: Raw Data File \n",
    "        # derived data file: Derived Data File\n",
    "        filename = \"a_assay_mirnaqrtpcr_sermas.txt\",\n",
    "        measurement_type = OntologyAnnotation(\n",
    "            #term = \"transcription profiling by RT-PCR assay\", \n",
    "            #term_source = ontologies[\"OBI\"],\n",
    "            #term_accession = \"http://purl.obolibrary.org/obo/OBI_0001361\"),\n",
    "            term = \"transcription profiling\", \n",
    "            term_source = ontologies[\"EFO\"],\n",
    "            term_accession = \"http://www.ebi.ac.uk/efo/EFO_0001032\"),\n",
    "        technology_type = OntologyAnnotation(\n",
    "            term = \"microRNA profiling by RT-PCR\", \n",
    "            term_source = ontologies[\"EFO\"], \n",
    "            term_accession = \"http://www.ebi.ac.uk/efo/EFO_0007687\"),\n",
    "        technology_platform = OntologyAnnotation(\n",
    "            term = \"LightCycler 480 Real-Time PCR detection instrument\",\n",
    "            term_source = ontologies[\"BAO\"], \n",
    "            term_accession = \"http://www.bioassayontology.org/bao#BAO_0150063\")\n",
    "    ),\n",
    "#    \"proteomics_imtm\": Assay(\n",
    "#        filename = \"a_assay_proteomics_imtm.txt\"), \n",
    "    \"metabolomics_acylcarnitines_rumc\": Assay(\n",
    "        # measurement type: targeted metabolite profiling\n",
    "        # technology type: mass spectrometry\n",
    "        # protocol type: mass spectrometry\n",
    "        # raw data file: Raw Spectral Data File\n",
    "        # derived data file:\n",
    "        #     - Derived Spectral Data File\n",
    "        #     - Metabolite Assignment File\n",
    "        # https://doi.org/10.1023/A:1005587617745\n",
    "        filename = \"a_assay_metabolomics_acylcarnitines_rumc.txt\",\n",
    "        measurement_type = OntologyAnnotation(\n",
    "            term = \"targeted metabolite profiling\",\n",
    "            term_source = ontologies[\"MSIO\"],\n",
    "            term_accession = \"http://purl.obolibrary.org/obo/MSIO_0000100\"),\n",
    "        technology_type = OntologyAnnotation(\n",
    "            term = \"mass spectrometry\", #\"electrospray ionisation tandem mass spectrometry\",\n",
    "            term_source = ontologies[\"CHMO\"], \n",
    "            term_accession = \"http://purl.obolibrary.org/obo/CHMO_0000470\"), #\"http://purl.obolibrary.org/obo/CHMO_0000577\"),\n",
    "        technology_platform = \"Micromass Quattro II triple-quadrupole mass spectrometer\"\n",
    "    ),\n",
    "    # http://snomed.info/id/442613004 Quantitative measurement of acylcarnitine in plasma specimen (procedure)\n",
    "    \"metabolomics_aminoacids_mumc\": Assay(\n",
    "        filename = \"a_assay_metabolomics_aminoacids_mumc.txt\",\n",
    "        measurement_type = OntologyAnnotation(\n",
    "            term = \"targeted metabolite profiling\",\n",
    "            term_source = ontologies[\"MSIO\"],\n",
    "            term_accession = \"http://purl.obolibrary.org/obo/MSIO_0000100\"),\n",
    "        technology_type = OntologyAnnotation(\n",
    "            term = \"mass spectrometry\", \n",
    "            term_source = ontologies[\"CHMO\"], \n",
    "            term_accession = \"http://purl.obolibrary.org/obo/CHMO_0000470\"),\n",
    "        technology_platform = \"Micromass Quattro Premier XE Tandem Mass Spectrometer\" # from # https://doi.org/10.1016/j.cca.2009.06.023\n",
    "    ), \n",
    "#    \"metabolomics_fattyacids_mumc\": Assay(\n",
    "#        filename = \"a_assay_metabolomics_fattyacids_mumc.txt\"), \n",
    "}\n",
    "# NOTE: make sure every assay contains samples etc. before added to study; \n",
    "#       otherwise it's possible that some assays are not written to files "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Define materials and processes referring to protocols for all assays\n",
    "\n",
    "Data file types are:\n",
    "'Raw Data File',\n",
    "'Derived Data File',\n",
    "'Image File',\n",
    "'Acquisition Parameter Data File',\n",
    "'Derived Spectral Data File',\n",
    "'Protein Assignment File',\n",
    "'Raw Spectral Data File',\n",
    "'Peptide Assignment File',\n",
    "'Array Data File',\n",
    "'Derived Array Data File',\n",
    "'Post Translational Modification Assignment File',\n",
    "'Derived Array Data Matrix File',\n",
    "'Free Induction Decay Data File',\n",
    "'Metabolite Assignment File',\n",
    "'Array Data Matrix File'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# genomics IMTM\n",
    "for idx, sample in enumerate(cohort_study.samples):\n",
    "    # DNA extraction\n",
    "    dna = Material(\n",
    "        name = \"DNA_{0}\".format(sample.name),\n",
    "        type_ = \"Extract Name\")\n",
    "    dna_extraction_process = Process(\n",
    "        name = \"DNA_extraction_{0}\".format(sample.name),\n",
    "        executes_protocol = protocols[\"dna_extraction\"],\n",
    "        inputs = [sample], \n",
    "        outputs = [dna])\n",
    "    ##################################\n",
    "    # genomics measurement\n",
    "    # TODO derived data files \n",
    "    wgs_raw_file1 = DataFile(\n",
    "        filename = \"WGS_rawdata_{0}_R1.fastq.gz\".format(dna.name), \n",
    "        label = \"Raw Data File R1\", \n",
    "        generated_from = [dna])\n",
    "    wgs_raw_file2 = DataFile(\n",
    "        filename = \"WGS_rawdata_{0}_R2.fastq.gz\".format(dna.name), \n",
    "        label = \"Raw Data File R2\", \n",
    "        generated_from = [dna])\n",
    "    wgs_process = Process(\n",
    "        name = \"WGS_{0}\".format(dna.name),\n",
    "        executes_protocol = protocols[\"Whole Genome Sequencing\"],\n",
    "        inputs = [dna], \n",
    "        outputs = [wgs_raw_file1, wgs_raw_file2])\n",
    "    plink(dna_extraction_process, wgs_process)\n",
    "    assays[\"genomics_imtm\"].samples.append(sample)\n",
    "    assays[\"genomics_imtm\"].other_material.append(dna)\n",
    "    assays[\"genomics_imtm\"].data_files.append(wgs_raw_file1)\n",
    "    assays[\"genomics_imtm\"].data_files.append(wgs_raw_file2)\n",
    "    assays[\"genomics_imtm\"].process_sequence.append(dna_extraction_process)\n",
    "    assays[\"genomics_imtm\"].process_sequence.append(wgs_process)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# methylation sequencing UU\n",
    "for idx, sample in enumerate(cohort_study.samples):\n",
    "    ################################\n",
    "    # materials used in multiple assays\n",
    "    # DNA extraction\n",
    "    dna = Material(\n",
    "        name = \"DNA_{0}\".format(sample.name),\n",
    "        type_ = \"Extract Name\")\n",
    "    dna_extraction_process = Process(\n",
    "        name = \"DNA_extraction_{0}\".format(sample.name),\n",
    "        executes_protocol = protocols[\"dna_extraction\"],\n",
    "        inputs = [sample], \n",
    "        outputs = [dna])\n",
    "    ############################\n",
    "    # methylation sequencing\n",
    "    # TODO issue - is this the correct way of handling two raw data files? Should be of type \"Raw Data File\", but there can only be one file of this type\n",
    "    methylseq_raw_file1 = DataFile(\n",
    "        filename = \"EMseq_rawdata_{0}_R1.fastq\".format(dna.name), \n",
    "        label = \"Raw Data File R1\", \n",
    "        generated_from = [dna])\n",
    "    methylseq_raw_file2 = DataFile(\n",
    "        filename = \"EMseq_rawdata_{0}_R2.fastq\".format(dna.name), \n",
    "        label = \"Raw Data File R2\", \n",
    "        generated_from = [dna])\n",
    "    methylseq_process = Process(\n",
    "        name = \"WGS_{0}\".format(dna.name),\n",
    "        executes_protocol = protocols[\"EM-seq\"],\n",
    "        inputs = [dna], \n",
    "        outputs = [methylseq_raw_file1, methylseq_raw_file2])\n",
    "    plink(dna_extraction_process, methylseq_process)\n",
    "    assays[\"methylseq_uu\"].samples.append(sample)\n",
    "    assays[\"methylseq_uu\"].other_material.append(dna)\n",
    "    assays[\"methylseq_uu\"].data_files.append(methylseq_raw_file1)\n",
    "    assays[\"methylseq_uu\"].data_files.append(methylseq_raw_file2)\n",
    "    assays[\"methylseq_uu\"].process_sequence.append(dna_extraction_process)\n",
    "    assays[\"methylseq_uu\"].process_sequence.append(methylseq_process)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# RNA-seq FIMM\n",
    "for idx, sample in enumerate(cohort_study.samples):\n",
    "    # RNA extraction\n",
    "    rna = Material(\n",
    "        name = \"RNA_{0}\".format(sample.name),\n",
    "        type_ = \"Extract Name\")\n",
    "    rna_extraction_process = Process(\n",
    "        name = \"RNA_extraction_{0}\".format(sample.name),\n",
    "        executes_protocol = protocols[\"rna_extraction\"],\n",
    "        inputs = [sample], \n",
    "        outputs = [rna])\n",
    "    ##################################\n",
    "    # transcriptomics - RNAseq\n",
    "    # TODO derived data files \n",
    "    rnaseq_raw_file1 = DataFile(\n",
    "        filename = \"RNAseq_rawdata_{0}_R1.fastq.gz\".format(rna.name), \n",
    "        label = \"Raw Data File R1\", \n",
    "        generated_from = [rna])\n",
    "    rnaseq_raw_file2 = DataFile(\n",
    "        filename = \"RNAseq_rawdata_{0}_R2.fastq.gz\".format(rna.name), \n",
    "        label = \"Raw Data File R2\", \n",
    "        generated_from = [rna])\n",
    "    rnaseq_libraryprep_process = Process(\n",
    "        name = \"mRNAlibprep_{0}\".format(rna.name),\n",
    "        executes_protocol = protocols[\"mRNA library preparation\"],\n",
    "        inputs = [rna], \n",
    "        outputs = [])\n",
    "    rnaseq_process = Process(\n",
    "        name = \"mRNAseq_{0}\".format(rna.name),\n",
    "        executes_protocol = protocols[\"mRNA Sequencing\"],\n",
    "        inputs = [], \n",
    "        outputs = [rnaseq_raw_file1, rnaseq_raw_file2])\n",
    "    plink(rna_extraction_process, rnaseq_libraryprep_process)\n",
    "    plink(rnaseq_libraryprep_process, rnaseq_process)\n",
    "    assays[\"rnaseq_fimm\"].samples.append(sample)\n",
    "    assays[\"rnaseq_fimm\"].other_material.append(rna)\n",
    "    assays[\"rnaseq_fimm\"].data_files.append(rnaseq_raw_file1)\n",
    "    assays[\"rnaseq_fimm\"].data_files.append(rnaseq_raw_file2)\n",
    "    assays[\"rnaseq_fimm\"].process_sequence.append(rna_extraction_process)\n",
    "    assays[\"rnaseq_fimm\"].process_sequence.append(rnaseq_libraryprep_process)\n",
    "    assays[\"rnaseq_fimm\"].process_sequence.append(rnaseq_process)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# miRNA qRT-PCR SERMAS\n",
    "for idx, sample in enumerate(cohort_study.samples):\n",
    "    # RNA extraction\n",
    "    rna = Material(\n",
    "        name = \"RNA_{0}\".format(sample.name),\n",
    "        type_ = \"Extract Name\")\n",
    "    rna_extraction_process = Process(\n",
    "        name = \"RNA_extraction_{0}\".format(sample.name),\n",
    "        executes_protocol = protocols[\"rna_extraction\"],\n",
    "        inputs = [sample], \n",
    "        outputs = [rna])\n",
    "    ##################################\n",
    "    # microRNA qRT-PCR - SERMAS\n",
    "    # 1 sample -> 1 RNA isolation -> 1 cDNA -> triplicate PCR\n",
    "    cdna = Material(\n",
    "        name = \"cDNA_{0}\".format(sample.name),\n",
    "        type_ = \"Labeled Extract Name\")\n",
    "    reversetranscription_process = Process(\n",
    "        name = \"ReverseTranscription_{0}\".format(rna.name),\n",
    "        executes_protocol = protocols[\"Reverse Transcription\"],\n",
    "        inputs = [rna],\n",
    "        outputs = [cdna])\n",
    "    for techrep in range(1,4): # triplicate PCR\n",
    "    #techrep = 1\n",
    "        miRNAqRTPCR_raw_file = DataFile(\n",
    "            filename = \"miRNAqPCR_rawdata_{0}_Replicate_{1}\".format(sample.name, techrep), \n",
    "            label = \"Raw Data File\", \n",
    "            generated_from = [cdna])\n",
    "        miRNAarray_process = Process(\n",
    "            name = \"miRNAarray_{0}_Replicate_{1}\".format(sample.name, techrep), \n",
    "            executes_protocol = protocols[\"miRNA array\"],\n",
    "            parameter_values = [\n",
    "                ParameterValue(category = protocol_params[\"technical replicate\"],\n",
    "                               value = techrep)],\n",
    "            inputs = [cdna], \n",
    "            outputs = [miRNAqRTPCR_raw_file])\n",
    "        miRNAqRTPCR_derived_file = DataFile(\n",
    "            filename = \"miRNAqPCR_normalizationdata_{0}_Replicate_{1}\".format(sample.name, techrep), \n",
    "            label = \"Derived Data File\", \n",
    "            generated_from = [miRNAqRTPCR_raw_file])\n",
    "        datanorm_process = Process(\n",
    "            name = \"datanormalization_{0}_Replicate_{1}\".format(sample.name, techrep), \n",
    "            executes_protocol = protocols[\"Data Normalization\"], \n",
    "            inputs = [miRNAqRTPCR_raw_file], \n",
    "            outputs = [miRNAqRTPCR_derived_file])\n",
    "\n",
    "        plink(rna_extraction_process, reversetranscription_process)\n",
    "        plink(reversetranscription_process, miRNAarray_process)\n",
    "        plink(miRNAarray_process, datanorm_process)\n",
    "\n",
    "        assays[\"mirnaqrtpcr_sermas\"].samples.append(sample)\n",
    "        assays[\"mirnaqrtpcr_sermas\"].other_material.append(rna)\n",
    "        assays[\"mirnaqrtpcr_sermas\"].other_material.append(cdna)\n",
    "\n",
    "        assays[\"mirnaqrtpcr_sermas\"].data_files.append(miRNAqRTPCR_raw_file)\n",
    "        assays[\"mirnaqrtpcr_sermas\"].data_files.append(miRNAqRTPCR_derived_file)\n",
    "\n",
    "        assays[\"mirnaqrtpcr_sermas\"].process_sequence.append(rna_extraction_process)\n",
    "        assays[\"mirnaqrtpcr_sermas\"].process_sequence.append(reversetranscription_process)\n",
    "        assays[\"mirnaqrtpcr_sermas\"].process_sequence.append(miRNAarray_process)\n",
    "        assays[\"mirnaqrtpcr_sermas\"].process_sequence.append(datanorm_process)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# metabolomics RUMC\n",
    "for idx, sample in enumerate(cohort_study.samples):\n",
    "    ##################################\n",
    "    # metabolomics_acylcarnitines_rumc\n",
    "    acylcarnitine_sample_extract = Material(\n",
    "        name = \"{0}^B-BCARP_NM\".format(sample.name),\n",
    "        type_ = \"Extract Name\")\n",
    "    acylcarnitine_sampleprep_process = Process(\n",
    "        name = \"Sample_preparation_plasma_acylcarnitine_analysis_{0}\".format(sample.name),\n",
    "        executes_protocol = protocols[\"Extraction\"],\n",
    "        inputs = [sample], \n",
    "        outputs = [acylcarnitine_sample_extract],\n",
    "        parameter_values = [\n",
    "            ParameterValue(category = protocol_params[\"Post Extraction\"],\n",
    "                           value = \"50 uL of sample was diluted with IS in MeOH, followed by 500 uL acetonitril. Mixture was vortexed and subsequently centrifuged at 16100*g at room temp. For 5 minutes. Supernatant was transferred to a new vial and evaporated at 37 degree Celsius under a flow of nitrogen. 50 uL of 1M derivitizing agent (1-butanol : acetylchloride 19:1) was added. The sample was vortexed and kept at 60° Celsius for exactly 15 minutes. Sample was again evaporated at 37° Celsius under a flow of nitrogen. Sample was then resuspended in 500 uL acetonitril, vortexed and injected.\"),\n",
    "            ParameterValue(category = protocol_params[\"Derivatization\"],\n",
    "                           value = \"Butylation\")])\n",
    "    acylcarnitine_chromatography_process = Process(\n",
    "        name = \"Chromatography_plasma_acylcarnitine_analysis_{0}\".format(sample.name),\n",
    "        executes_protocol = protocols[\"Chromatography\"],\n",
    "        inputs = [acylcarnitine_sample_extract], \n",
    "        outputs = [],\n",
    "        parameter_values = [\n",
    "            ParameterValue(category = protocol_params[\"Chromatography Instrument\"],\n",
    "                           value = \"Waters I-class HPLC system\"),\n",
    "            ParameterValue(category = protocol_params[\"Column type\"],\n",
    "                           value = \"flow injection analysis\")])\n",
    "    acylcarnitine_raw_data = DataFile(\n",
    "        filename = \"{0}^B-BCARP_NM.raw\".format(sample.name), \n",
    "        label = \"Raw Spectral Data File\")\n",
    "    acylcarnitine_MS_process = Process(\n",
    "        name = \"MS_plasma_acylcarnitine_analysis_{0}\".format(sample.name),\n",
    "        executes_protocol = protocols[\"Mass spectrometry\"],\n",
    "        inputs = [], \n",
    "        outputs = [acylcarnitine_raw_data],\n",
    "        parameter_values = [\n",
    "            ParameterValue(category = protocol_params[\"Scan polarity\"],\n",
    "                           value = \"postive\"),\n",
    "            ParameterValue(category = protocol_params[\"Scan m/z range\"],\n",
    "                           value = \"200-525\"),\n",
    "            ParameterValue(category = protocol_params[\"Instrument\"],\n",
    "                           value = \"Waters Xevo TQ-S micro\"),\n",
    "            ParameterValue(category = protocol_params[\"Ion source\"],\n",
    "                           value = OntologyAnnotation(\n",
    "                               term = \"electrospray ionization\",\n",
    "                               term_source = ontologies[\"CHMO\"],\n",
    "                               term_accession = \"http://purl.obolibrary.org/obo/MS_1000073\")), \n",
    "            ParameterValue(category = protocol_params[\"Mass analyzer\"],\n",
    "                           value = OntologyAnnotation(\n",
    "                               term = \"triple quadrupole mass spectrometer\",\n",
    "                               term_source = ontologies[\"CHMO\"],\n",
    "                               term_accession = \"http://purl.obolibrary.org/obo/CHMO_0002021\")),\n",
    "            ParameterValue(category = protocol_params[\"method reference\"],\n",
    "                           value = \"https://doi.org/10.1023/A:1005587617745\")])\n",
    "    acylcarnitine_datatransform_process = Process(\n",
    "        name = \"Data_transformation_plasma_acylcarnitine_analysis_{0}\".format(sample.name),\n",
    "        inputs = [acylcarnitine_raw_data],\n",
    "        outputs = [],\n",
    "        executes_protocol = protocols[\"Data transformation_acylcarnitines\"])\n",
    "    acylcarnitine_MAF = DataFile(\n",
    "        filename = \"m_carnprofiel.tsv\", \n",
    "        label = \"Metabolite Assignment File\")\n",
    "    acylcarnitine_metaboliteident_process = Process(\n",
    "        name = \"Metabolite_identification_plasma_acylcarnitine_analysis_{0}\".format(sample.name),\n",
    "        outputs = [acylcarnitine_MAF],\n",
    "        executes_protocol = protocols[\"Metabolite identification_acylcarnitines\"])\n",
    "    plink(acylcarnitine_sampleprep_process, acylcarnitine_chromatography_process)\n",
    "    plink(acylcarnitine_chromatography_process, acylcarnitine_MS_process)\n",
    "    #plink(acylcarnitine_sampleprep_process, acylcarnitine_MS_process)\n",
    "    plink(acylcarnitine_MS_process, acylcarnitine_datatransform_process)\n",
    "    plink(acylcarnitine_datatransform_process, acylcarnitine_metaboliteident_process)\n",
    "    assays[\"metabolomics_acylcarnitines_rumc\"].samples.append(sample)\n",
    "    assays[\"metabolomics_acylcarnitines_rumc\"].other_material.append(acylcarnitine_sample_extract)\n",
    "    assays[\"metabolomics_acylcarnitines_rumc\"].process_sequence.append(acylcarnitine_sampleprep_process)\n",
    "    assays[\"metabolomics_acylcarnitines_rumc\"].process_sequence.append(acylcarnitine_chromatography_process)\n",
    "    assays[\"metabolomics_acylcarnitines_rumc\"].process_sequence.append(acylcarnitine_MS_process)\n",
    "    assays[\"metabolomics_acylcarnitines_rumc\"].process_sequence.append(acylcarnitine_datatransform_process)\n",
    "    assays[\"metabolomics_acylcarnitines_rumc\"].process_sequence.append(acylcarnitine_metaboliteident_process)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "genomics_imtm\n",
      "methylseq_uu\n",
      "rnaseq_fimm\n",
      "mirnaseq_fimm\n",
      "mirnaqrtpcr_sermas\n",
      "metabolomics_acylcarnitines_rumc\n",
      "metabolomics_aminoacids_mumc\n"
     ]
    }
   ],
   "source": [
    "# add assays to study\n",
    "assay_filenames = []\n",
    "for assay_name, assay in assays.items():\n",
    "    print(assay_name)\n",
    "    if assay_name in (\"metabolomics_acylcarnitines_rumc\", \"mirnaqrtpcr_sermas\", \n",
    "                      \"rnaseq_fimm\", \"genomics_imtm\", \"methylseq_uu\"):\n",
    "        cohort_study.assays.append(assay)\n",
    "        assay_filenames.append(assay.filename)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "#print(cohort_study.assays)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Export to ISA-Tab and ISA-JSON"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Write ISA-Tab files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "isatools.model.Investigation(identifier='', filename='i_investigation.txt', title='EATRIS-Plus - Flagship in Personalised Medicine', submission_date='', public_release_date='', ontology_source_references=[isatools.model.OntologySource(name='BAO - BioAssay Ontology', file='http://www.bioassayontology.org/bao/bao_complete.owl', version='', description='The BioAssay Ontology (BAO) describes biological screening assays and their results including high-throughput screening (HTS) data for the purpose of categorizing assays and data analysis. BAO is an extensible, knowledge-based, highly expressive (currently SHOIQ(D)) description of biological assays making use of descriptive logic based features of the Web Ontology Language (OWL). BAO currently has over 700 classes and also makes use of several other ontologies. It describes several concepts related to biological screening, including Perturbagen, Format, Meta Target, Design, Detection Technology, and Endpoint. Perturbagens are perturbing agents that are screened in an assay; they are mostly small molecules. Assay Meta Target describes what is known about the biological system and / or its components interrogated in the assay (and influenced by the Perturbagen). Meta target can be directly described as a molecular entity (e.g. a purified protein or a protein complex), or indirectly by a biological process or event (e.g. phosphorylation). Format describes the biological or chemical features common to each test condition in the assay and includes biochemical, cell-based, organism-based, and variations thereof. The assay Design describes the assay methodology and implementation of how the perturbation of the biological system is translated into a detectable signal. Detection Technology relates to the physical method and technical details to detect and record a signal. Endpoints are the final HTS results as they are usually published (such as IC50, percent inhibition, etc). BAO has been designed to accommodate multiplexed assays. All main BAO components include multiple levels of sub-categories and specification classes, which are linked via object property relationships forming an expressive knowledge-based representation.', comments=[]), isatools.model.OntologySource(name='CHEBI - Chemical Entities of Biological Interest', file='http://purl.obolibrary.org/obo/chebi.owl', version='', description='A structured classification of molecular entities of biological interest focusing on 'small' chemical compounds.', comments=[]), isatools.model.OntologySource(name='CHMO - Chemical Methods Ontology', file='http://purl.obolibrary.org/obo/chmo.owl', version='', description='CHMO, the chemical methods ontology, describes methods used to collect data in chemical experiments, such as mass spectrometry and electron microscopy prepare and separate material for further analysis, such as sample ionisation, chromatography, and electrophoresis synthesise materials, such as epitaxy and continuous vapour deposition It also describes the instruments used in these experiments, such as mass spectrometers and chromatography columns. It is intended to be complementary to the Ontology for Biomedical Investigations (OBI).', comments=[]), isatools.model.OntologySource(name='CRO - Contributor Role Ontology', file='http://purl.obolibrary.org/obo/cro.owl', version='', description='A classification of the diverse roles performed in the work leading to a published research output in the sciences. Its purpose to provide transparency in contributions to scholarly published work, to enable improved systems of attribution, credit, and accountability.', comments=[]), isatools.model.OntologySource(name='EDAM - EMBRACE Data and Methods', file='http://edamontology.org/EDAM.owl', version='', description='EDAM (EMBRACE Data and Methods) is an ontology of common bioinformatics operations, topics, types of data including identifiers, and formats. EDAM comprises common concepts (shared within the bioinformatics community) that apply to semantic annotation of resources.', comments=[]), isatools.model.OntologySource(name='EFO - Experimental Factor Ontology', file='http://www.ebi.ac.uk/efo/efo.owl', version='', description='The Experimental Factor Ontology (EFO) provides a systematic description of many experimental variables available in EBI databases, and for external projects such as the NHGRI GWAS catalogue. It combines parts of several biological ontologies, such as anatomy, disease and chemical compounds. The scope of EFO is to support the annotation, analysis and visualization of data handled by many groups at the EBI and as the core ontology for OpenTargets.org', comments=[]), isatools.model.OntologySource(name='GENEPIO - Genomic Epidemiology Ontology', file='http://purl.obolibrary.org/obo/genepio.owl', version='', description='The Genomic Epidemiology Ontology (GenEpiO) covers vocabulary necessary to identify, document and research foodborne pathogens and associated outbreaks.', comments=[]), isatools.model.OntologySource(name='MI - Molecular Interactions Controlled Vocabulary', file='http://purl.obolibrary.org/obo/mi.owl', version='', description='A structured controlled vocabulary for the annotation of experiments concerned with protein-protein interactions.', comments=[]), isatools.model.OntologySource(name='MMO - Measurement method ontology', file='http://purl.obolibrary.org/obo/mmo.owl', version='', description='A representation of the variety of methods used to make clinical and phenotype measurements.', comments=[]), isatools.model.OntologySource(name='Metabolomics Standards Initiative Ontology (MSIO)', file='http://purl.obolibrary.org/obo/ms.owl', version='', description='A structured controlled vocabulary for the annotation of experiments concerned with proteomics mass spectrometry.', comments=[]), isatools.model.OntologySource(name='MS - Mass spectrometry ontology', file='http://purl.obolibrary.org/obo/msio.owl', version='', description='MSIO aims to provide a single point of entry to support semantic markup of experiments making use of NMR and MS techniques to identify, measure and quantify small molecules known as metabolites. MSIO covers metabolite profiling, targeted or undertargeted, tracer based applications. MSIO reuses a number of resources such as CHEBI, DUO, NMRCV, OBI, and STATO.', comments=[]), isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), isatools.model.OntologySource(name='NCI Thesaurus OBO Edition', file='http://purl.obolibrary.org/obo/ncit.owl', version='', description='The NCIt OBO Edition project aims to increase integration of the NCIt with OBO Library ontologies. NCIt is a reference terminology that includes broad coverage of the cancer domain, including cancer related diseases, findings and abnormalities. NCIt OBO Edition releases should be considered experimental.', comments=[]), isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), isatools.model.OntologySource(name='Ontologized MIABIS', file='http://purl.obolibrary.org/obo/omiabis.owl', version='', description='An ontological version of MIABIS (Minimum Information About BIobank data Sharing)', comments=[]), isatools.model.OntologySource(name='Ontology for RNA sequencing (ORNASEQ)', file='http://purl.obolibrary.org/obo/ornaseq.owl', version='', description='An application ontology designed to annotate next-generation sequencing experiments performed on RNA.', comments=[]), isatools.model.OntologySource(name='PRIDE Controlled Vocabulary', file='http://purl.obolibrary.org/obo/pride_cv.obo', version='', description='The PRIDE PRoteomics IDEntifications (PRIDE) database is a centralized, standards compliant, public data repository for proteomics data, including protein and peptide identifications, post-translational modifications and supporting spectral evidence.', comments=[]), isatools.model.OntologySource(name='STATO: the statistical methods ontology', file='http://purl.obolibrary.org/obo/stato.owl', version='', description='STATO is the statistical methods ontology. It contains concepts and properties related to statistical methods, probability distributions and other concepts related to statistical analysis, including relationships to study designs and plots.', comments=[]), isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), isatools.model.OntologySource(name='to be defined', file='http://tbd.owl', version='', description='tbd', comments=[])], publications=[], contacts=[isatools.model.Person(last_name='Keidong', first_name='Eliis', mid_initials='', email='', phone='', fax='', address='', affiliation='EATRIS', roles=[isatools.model.OntologyAnnotation(term='project management role', term_source=isatools.model.OntologySource(name='CRO - Contributor Role Ontology', file='http://purl.obolibrary.org/obo/cro.owl', version='', description='A classification of the diverse roles performed in the work leading to a published research output in the sciences. Its purpose to provide transparency in contributions to scholarly published work, to enable improved systems of attribution, credit, and accountability.', comments=[]), term_accession='http://purl.obolibrary.org/obo/CRO_0000065', comments=[])], comments=[])], studies=[isatools.model.Study(filename='s_study.txt', identifier='', title='Multi-omics data of a Czech population cohort', description='Multi-omics data of a Czech population cohort', submission_date='', public_release_date='', contacts=[isatools.model.Person(last_name='Hajduch', first_name='Marian', mid_initials='', email='', phone='', fax='', address='', affiliation='Institute of Molecular and Translational Medicine (IMTM), Palacky University Olomouc', roles=[], comments=[])], design_descriptors=[isatools.model.OntologyAnnotation(term='Multi-omics study', term_source=isatools.model.OntologySource(name='PRIDE Controlled Vocabulary', file='http://purl.obolibrary.org/obo/pride_cv.obo', version='', description='The PRIDE PRoteomics IDEntifications (PRIDE) database is a centralized, standards compliant, public data repository for proteomics data, including protein and peptide identifications, post-translational modifications and supporting spectral evidence.', comments=[]), term_accession='http://purl.obolibrary.org/obo/PRIDE_0000461', comments=[]), isatools.model.OntologyAnnotation(term='population based study design', term_source=isatools.model.OntologySource(name='Ontologized MIABIS', file='http://purl.obolibrary.org/obo/omiabis.owl', version='', description='An ontological version of MIABIS (Minimum Information About BIobank data Sharing)', comments=[]), term_accession='http://purl.obolibrary.org/obo/OMIABIS_0001022', comments=[])], publications=[], factors=[], protocols=[isatools.model.Protocol(name='sample_collection_protocol', protocol_type=isatools.model.OntologyAnnotation(term='sample collection', term_source=None, term_accession='None', comments=[]), uri='', version='', parameters=[], components=[], comments=[]), isatools.model.Protocol(name='DNA extraction', protocol_type=isatools.model.OntologyAnnotation(term='DNA extraction', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0000257', comments=[]), uri='', version='', parameters=[], components=[], comments=[]), isatools.model.Protocol(name='RNA extraction', protocol_type=isatools.model.OntologyAnnotation(term='RNA extraction', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0666666', comments=[]), uri='', version='', parameters=[], components=[], comments=[]), isatools.model.Protocol(name='Whole Genome Sequencing', protocol_type=isatools.model.OntologyAnnotation(term='Whole Genome Sequencing', term_source=isatools.model.OntologySource(name='NCI Thesaurus OBO Edition', file='http://purl.obolibrary.org/obo/ncit.owl', version='', description='The NCIt OBO Edition project aims to increase integration of the NCIt with OBO Library ontologies. NCIt is a reference terminology that includes broad coverage of the cancer domain, including cancer related diseases, findings and abnormalities. NCIt OBO Edition releases should be considered experimental.', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCIT_C101294', comments=[]), uri='', version='', parameters=[], components=[], comments=[]), isatools.model.Protocol(name='mRNA library preparation', protocol_type=isatools.model.OntologyAnnotation(term='library preparation protocol', term_source=isatools.model.OntologySource(name='Ontology for RNA sequencing (ORNASEQ)', file='http://purl.obolibrary.org/obo/ornaseq.owl', version='', description='An application ontology designed to annotate next-generation sequencing experiments performed on RNA.', comments=[]), term_accession='http://purl.obolibrary.org/obo/ORNASEQ_0000007', comments=[]), uri='https://www.illumina.com/products/by-type/sequencing-kits/library-prep-kits/stranded-mrna-prep.html', version='', parameters=[], components=[], comments=[]), isatools.model.Protocol(name='mRNA Sequencing', protocol_type=isatools.model.OntologyAnnotation(term='nucleic acid sequencing', term_source=isatools.model.OntologySource(name='NCI Thesaurus OBO Edition', file='http://purl.obolibrary.org/obo/ncit.owl', version='', description='The NCIt OBO Edition project aims to increase integration of the NCIt with OBO Library ontologies. NCIt is a reference terminology that includes broad coverage of the cancer domain, including cancer related diseases, findings and abnormalities. NCIt OBO Edition releases should be considered experimental.', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCIT_C18881', comments=[]), uri='', version='', parameters=[], components=[], comments=[]), isatools.model.Protocol(name='MicroRNA Sequencing', protocol_type=isatools.model.OntologyAnnotation(term='MicroRNA Sequencing', term_source=isatools.model.OntologySource(name='NCI Thesaurus OBO Edition', file='http://purl.obolibrary.org/obo/ncit.owl', version='', description='The NCIt OBO Edition project aims to increase integration of the NCIt with OBO Library ontologies. NCIt is a reference terminology that includes broad coverage of the cancer domain, including cancer related diseases, findings and abnormalities. NCIt OBO Edition releases should be considered experimental.', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCIT_C156057', comments=[]), uri='', version='', parameters=[], components=[], comments=[]), isatools.model.Protocol(name='Reverse Transcription', protocol_type=isatools.model.OntologyAnnotation(term='artificially induced reverse transcription', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0600028', comments=[]), uri='', version='', parameters=[], components=[], comments=[]), isatools.model.Protocol(name='miRNA array', protocol_type=isatools.model.OntologyAnnotation(term='microRNA profiling by RT-PCR', term_source=isatools.model.OntologySource(name='EFO - Experimental Factor Ontology', file='http://www.ebi.ac.uk/efo/efo.owl', version='', description='The Experimental Factor Ontology (EFO) provides a systematic description of many experimental variables available in EBI databases, and for external projects such as the NHGRI GWAS catalogue. It combines parts of several biological ontologies, such as anatomy, disease and chemical compounds. The scope of EFO is to support the annotation, analysis and visualization of data handled by many groups at the EBI and as the core ontology for OpenTargets.org', comments=[]), term_accession='http://www.ebi.ac.uk/efo/EFO_0007687', comments=[]), uri='', version='', parameters=[isatools.model.ProtocolParameter(parameter_name=isatools.model.OntologyAnnotation(term='technical replicate', term_source=isatools.model.OntologySource(name='EFO - Experimental Factor Ontology', file='http://www.ebi.ac.uk/efo/efo.owl', version='', description='The Experimental Factor Ontology (EFO) provides a systematic description of many experimental variables available in EBI databases, and for external projects such as the NHGRI GWAS catalogue. It combines parts of several biological ontologies, such as anatomy, disease and chemical compounds. The scope of EFO is to support the annotation, analysis and visualization of data handled by many groups at the EBI and as the core ontology for OpenTargets.org', comments=[]), term_accession='http://www.ebi.ac.uk/efo/EFO_0002090', comments=[]), comments=[])], components=[], comments=[]), isatools.model.Protocol(name='Data Normalization', protocol_type=isatools.model.OntologyAnnotation(term='Constant Normalization', term_source=isatools.model.OntologySource(name='NCI Thesaurus OBO Edition', file='http://purl.obolibrary.org/obo/ncit.owl', version='', description='The NCIt OBO Edition project aims to increase integration of the NCIt with OBO Library ontologies. NCIt is a reference terminology that includes broad coverage of the cancer domain, including cancer related diseases, findings and abnormalities. NCIt OBO Edition releases should be considered experimental.', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCIT_C63786', comments=[]), uri='', version='', parameters=[], components=[], comments=[]), isatools.model.Protocol(name='Extraction', protocol_type=isatools.model.OntologyAnnotation(term='extraction', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0302884', comments=[]), uri='', version='', parameters=[isatools.model.ProtocolParameter(parameter_name=isatools.model.OntologyAnnotation(term='Post Extraction', term_source=None, term_accession='', comments=[]), comments=[]), isatools.model.ProtocolParameter(parameter_name=isatools.model.OntologyAnnotation(term='Derivatization', term_source=isatools.model.OntologySource(name='MS - Mass spectrometry ontology', file='http://purl.obolibrary.org/obo/msio.owl', version='', description='MSIO aims to provide a single point of entry to support semantic markup of experiments making use of NMR and MS techniques to identify, measure and quantify small molecules known as metabolites. MSIO covers metabolite profiling, targeted or undertargeted, tracer based applications. MSIO reuses a number of resources such as CHEBI, DUO, NMRCV, OBI, and STATO.', comments=[]), term_accession='http://purl.obolibrary.org/obo/MSIO_0000111', comments=[]), comments=[])], components=[], comments=[]), isatools.model.Protocol(name='Chromatography', protocol_type=isatools.model.OntologyAnnotation(term='chromatography', term_source=isatools.model.OntologySource(name='CHMO - Chemical Methods Ontology', file='http://purl.obolibrary.org/obo/chmo.owl', version='', description='CHMO, the chemical methods ontology, describes methods used to collect data in chemical experiments, such as mass spectrometry and electron microscopy prepare and separate material for further analysis, such as sample ionisation, chromatography, and electrophoresis synthesise materials, such as epitaxy and continuous vapour deposition It also describes the instruments used in these experiments, such as mass spectrometers and chromatography columns. It is intended to be complementary to the Ontology for Biomedical Investigations (OBI).', comments=[]), term_accession='http://purl.obolibrary.org/obo/CHMO_0001000', comments=[]), uri='', version='', parameters=[isatools.model.ProtocolParameter(parameter_name=isatools.model.OntologyAnnotation(term='Chromatography Instrument', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0000485', comments=[]), comments=[]), isatools.model.ProtocolParameter(parameter_name=isatools.model.OntologyAnnotation(term='Column model', term_source=None, term_accession='', comments=[]), comments=[]), isatools.model.ProtocolParameter(parameter_name=isatools.model.OntologyAnnotation(term='Column type', term_source=None, term_accession='', comments=[]), comments=[])], components=[], comments=[]), isatools.model.Protocol(name='Mass spectrometry', protocol_type=isatools.model.OntologyAnnotation(term='mass spectrometry', term_source=isatools.model.OntologySource(name='CHMO - Chemical Methods Ontology', file='http://purl.obolibrary.org/obo/chmo.owl', version='', description='CHMO, the chemical methods ontology, describes methods used to collect data in chemical experiments, such as mass spectrometry and electron microscopy prepare and separate material for further analysis, such as sample ionisation, chromatography, and electrophoresis synthesise materials, such as epitaxy and continuous vapour deposition It also describes the instruments used in these experiments, such as mass spectrometers and chromatography columns. It is intended to be complementary to the Ontology for Biomedical Investigations (OBI).', comments=[]), term_accession='http://purl.obolibrary.org/obo/CHMO_0000470', comments=[]), uri='', version='', parameters=[isatools.model.ProtocolParameter(parameter_name=isatools.model.OntologyAnnotation(term='scan polarity', term_source=isatools.model.OntologySource(name='Metabolomics Standards Initiative Ontology (MSIO)', file='http://purl.obolibrary.org/obo/ms.owl', version='', description='A structured controlled vocabulary for the annotation of experiments concerned with proteomics mass spectrometry.', comments=[]), term_accession='http://purl.obolibrary.org/obo/MS_1000465', comments=[]), comments=[]), isatools.model.ProtocolParameter(parameter_name=isatools.model.OntologyAnnotation(term='Scan m/z range', term_source=None, term_accession='', comments=[]), comments=[]), isatools.model.ProtocolParameter(parameter_name=isatools.model.OntologyAnnotation(term='Instrument', term_source=isatools.model.OntologySource(name='Metabolomics Standards Initiative Ontology (MSIO)', file='http://purl.obolibrary.org/obo/ms.owl', version='', description='A structured controlled vocabulary for the annotation of experiments concerned with proteomics mass spectrometry.', comments=[]), term_accession='http://purl.obolibrary.org/obo/MS_1000463', comments=[]), comments=[]), isatools.model.ProtocolParameter(parameter_name=isatools.model.OntologyAnnotation(term='Ion source', term_source=isatools.model.OntologySource(name='CHMO - Chemical Methods Ontology', file='http://purl.obolibrary.org/obo/chmo.owl', version='', description='CHMO, the chemical methods ontology, describes methods used to collect data in chemical experiments, such as mass spectrometry and electron microscopy prepare and separate material for further analysis, such as sample ionisation, chromatography, and electrophoresis synthesise materials, such as epitaxy and continuous vapour deposition It also describes the instruments used in these experiments, such as mass spectrometers and chromatography columns. It is intended to be complementary to the Ontology for Biomedical Investigations (OBI).', comments=[]), term_accession='http://purl.obolibrary.org/obo/CHMO_0000960', comments=[]), comments=[]), isatools.model.ProtocolParameter(parameter_name=isatools.model.OntologyAnnotation(term='Mass analyzer', term_source=isatools.model.OntologySource(name='Metabolomics Standards Initiative Ontology (MSIO)', file='http://purl.obolibrary.org/obo/ms.owl', version='', description='A structured controlled vocabulary for the annotation of experiments concerned with proteomics mass spectrometry.', comments=[]), term_accession='http://purl.obolibrary.org/obo/MS_1000451', comments=[]), comments=[]), isatools.model.ProtocolParameter(parameter_name=isatools.model.OntologyAnnotation(term='method reference', term_source=isatools.model.OntologySource(name='MI - Molecular Interactions Controlled Vocabulary', file='http://purl.obolibrary.org/obo/mi.owl', version='', description='A structured controlled vocabulary for the annotation of experiments concerned with protein-protein interactions.', comments=[]), term_accession='http://purl.obolibrary.org/obo/MI_0357', comments=[]), comments=[])], components=[], comments=[]), isatools.model.Protocol(name='Data transformation - Plasma acylcarnitine analysis', protocol_type=isatools.model.OntologyAnnotation(term='data transformation', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0200000', comments=[]), uri='https://doi.org/10.1023/A:1005587617745', version='', parameters=[], components=[], comments=[]), isatools.model.Protocol(name='Metabolite identification - Plasma acylcarnitine analysis', protocol_type=isatools.model.OntologyAnnotation(term='metabolite identification', term_source=isatools.model.OntologySource(name='MI - Molecular Interactions Controlled Vocabulary', file='http://purl.obolibrary.org/obo/mi.owl', version='', description='A structured controlled vocabulary for the annotation of experiments concerned with protein-protein interactions.', comments=[]), term_accession='http://purl.obolibrary.org/obo/MI_2131', comments=[]), uri='', version='', parameters=[], components=[], comments=[]), isatools.model.Protocol(name='EM-seq', protocol_type=isatools.model.OntologyAnnotation(term='methylation profiling by high throughput sequencing', term_source=isatools.model.OntologySource(name='EFO - Experimental Factor Ontology', file='http://www.ebi.ac.uk/efo/efo.owl', version='', description='The Experimental Factor Ontology (EFO) provides a systematic description of many experimental variables available in EBI databases, and for external projects such as the NHGRI GWAS catalogue. It combines parts of several biological ontologies, such as anatomy, disease and chemical compounds. The scope of EFO is to support the annotation, analysis and visualization of data handled by many groups at the EBI and as the core ontology for OpenTargets.org', comments=[]), term_accession='http://www.ebi.ac.uk/efo/EFO_0002761', comments=[]), uri='', version='', parameters=[], components=[], comments=[])], assays=[isatools.model.Assay(measurement_type=isatools.model.OntologyAnnotation(term='Whole Genome Sequencing', term_source=isatools.model.OntologySource(name='NCI Thesaurus OBO Edition', file='http://purl.obolibrary.org/obo/ncit.owl', version='', description='The NCIt OBO Edition project aims to increase integration of the NCIt with OBO Library ontologies. NCIt is a reference terminology that includes broad coverage of the cancer domain, including cancer related diseases, findings and abnormalities. NCIt OBO Edition releases should be considered experimental.', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCIT_C101294', comments=[]), technology_type=isatools.model.OntologyAnnotation(term='Nucleic Acid Sequencing', term_source=isatools.model.OntologySource(name='NCI Thesaurus OBO Edition', file='http://purl.obolibrary.org/obo/ncit.owl', version='', description='The NCIt OBO Edition project aims to increase integration of the NCIt with OBO Library ontologies. NCIt is a reference terminology that includes broad coverage of the cancer domain, including cancer related diseases, findings and abnormalities. NCIt OBO Edition releases should be considered experimental.', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCIT_C18881', comments=[]), technology_platform='OntologyAnnotation(\n",
       "    term=Illumina platform\n",
       "    term_source=GENEPIO - Genomic Epidemiology Ontology\n",
       "    term_accession=http://purl.obolibrary.org/obo/GENEPIO_0001923\n",
       "    comments=0 Comment objects\n",
       ")', filename='a_assay_genomics_imtm.txt', data_files=[isatools.model.DataFile(filename='WGS_rawdata_DNA_sample_1_R1.fastq.gz', label='Raw Data File R1', generated_from=[<isatools.model.Material object at 0x7eff946d2940>], comments=[]), isatools.model.DataFile(filename='WGS_rawdata_DNA_sample_1_R2.fastq.gz', label='Raw Data File R2', generated_from=[<isatools.model.Material object at 0x7eff946d2940>], comments=[]), isatools.model.DataFile(filename='WGS_rawdata_DNA_sample_2_R1.fastq.gz', label='Raw Data File R1', generated_from=[<isatools.model.Material object at 0x7eff946d2910>], comments=[]), isatools.model.DataFile(filename='WGS_rawdata_DNA_sample_2_R2.fastq.gz', label='Raw Data File R2', generated_from=[<isatools.model.Material object at 0x7eff946d2910>], comments=[]), isatools.model.DataFile(filename='WGS_rawdata_DNA_sample_3_R1.fastq.gz', label='Raw Data File R1', generated_from=[<isatools.model.Material object at 0x7eff946d27f0>], comments=[]), isatools.model.DataFile(filename='WGS_rawdata_DNA_sample_3_R2.fastq.gz', label='Raw Data File R2', generated_from=[<isatools.model.Material object at 0x7eff946d27f0>], comments=[])], samples=[isatools.model.Sample(name='sample_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[]), isatools.model.Sample(name='sample_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[]), isatools.model.Sample(name='sample_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[])], process_sequence=[isatools.model.Process(id=\"\". name=\"DNA_extraction_sample_1\", executes_protocol=Protocol(\n",
       "    name=DNA extraction\n",
       "    protocol_type=DNA extraction\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Sample(name='sample_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[])], outputs=[<isatools.model.Material object at 0x7eff946d2940>]), isatools.model.Process(id=\"\". name=\"WGS_DNA_sample_1\", executes_protocol=Protocol(\n",
       "    name=Whole Genome Sequencing\n",
       "    protocol_type=Whole Genome Sequencing\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7eff946d2940>], outputs=[isatools.model.DataFile(filename='WGS_rawdata_DNA_sample_1_R1.fastq.gz', label='Raw Data File R1', generated_from=[<isatools.model.Material object at 0x7eff946d2940>], comments=[]), isatools.model.DataFile(filename='WGS_rawdata_DNA_sample_1_R2.fastq.gz', label='Raw Data File R2', generated_from=[<isatools.model.Material object at 0x7eff946d2940>], comments=[])]), isatools.model.Process(id=\"\". name=\"DNA_extraction_sample_2\", executes_protocol=Protocol(\n",
       "    name=DNA extraction\n",
       "    protocol_type=DNA extraction\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Sample(name='sample_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[])], outputs=[<isatools.model.Material object at 0x7eff946d2910>]), isatools.model.Process(id=\"\". name=\"WGS_DNA_sample_2\", executes_protocol=Protocol(\n",
       "    name=Whole Genome Sequencing\n",
       "    protocol_type=Whole Genome Sequencing\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7eff946d2910>], outputs=[isatools.model.DataFile(filename='WGS_rawdata_DNA_sample_2_R1.fastq.gz', label='Raw Data File R1', generated_from=[<isatools.model.Material object at 0x7eff946d2910>], comments=[]), isatools.model.DataFile(filename='WGS_rawdata_DNA_sample_2_R2.fastq.gz', label='Raw Data File R2', generated_from=[<isatools.model.Material object at 0x7eff946d2910>], comments=[])]), isatools.model.Process(id=\"\". name=\"DNA_extraction_sample_3\", executes_protocol=Protocol(\n",
       "    name=DNA extraction\n",
       "    protocol_type=DNA extraction\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Sample(name='sample_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[])], outputs=[<isatools.model.Material object at 0x7eff946d27f0>]), isatools.model.Process(id=\"\". name=\"WGS_DNA_sample_3\", executes_protocol=Protocol(\n",
       "    name=Whole Genome Sequencing\n",
       "    protocol_type=Whole Genome Sequencing\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7eff946d27f0>], outputs=[isatools.model.DataFile(filename='WGS_rawdata_DNA_sample_3_R1.fastq.gz', label='Raw Data File R1', generated_from=[<isatools.model.Material object at 0x7eff946d27f0>], comments=[]), isatools.model.DataFile(filename='WGS_rawdata_DNA_sample_3_R2.fastq.gz', label='Raw Data File R2', generated_from=[<isatools.model.Material object at 0x7eff946d27f0>], comments=[])])], other_material=[<isatools.model.Material object at 0x7eff946d2940>, <isatools.model.Material object at 0x7eff946d2910>, <isatools.model.Material object at 0x7eff946d27f0>], characteristic_categories=[], comments=[], units=[]), isatools.model.Assay(measurement_type=isatools.model.OntologyAnnotation(term='DNA methylation profiling by high throughput sequencing assay', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0001266', comments=[]), technology_type=isatools.model.OntologyAnnotation(term='Nucleic Acid Sequencing', term_source=isatools.model.OntologySource(name='NCI Thesaurus OBO Edition', file='http://purl.obolibrary.org/obo/ncit.owl', version='', description='The NCIt OBO Edition project aims to increase integration of the NCIt with OBO Library ontologies. NCIt is a reference terminology that includes broad coverage of the cancer domain, including cancer related diseases, findings and abnormalities. NCIt OBO Edition releases should be considered experimental.', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCIT_C18881', comments=[]), technology_platform='NEBNext Enzymatic Methyl-seq (EM-seq)', filename='a_assay_methylseq_uu.txt', data_files=[isatools.model.DataFile(filename='EMseq_rawdata_DNA_sample_1_R1.fastq', label='Raw Data File R1', generated_from=[<isatools.model.Material object at 0x7eff946d2ac0>], comments=[]), isatools.model.DataFile(filename='EMseq_rawdata_DNA_sample_1_R2.fastq', label='Raw Data File R2', generated_from=[<isatools.model.Material object at 0x7eff946d2ac0>], comments=[]), isatools.model.DataFile(filename='EMseq_rawdata_DNA_sample_2_R1.fastq', label='Raw Data File R1', generated_from=[<isatools.model.Material object at 0x7eff517fd0a0>], comments=[]), isatools.model.DataFile(filename='EMseq_rawdata_DNA_sample_2_R2.fastq', label='Raw Data File R2', generated_from=[<isatools.model.Material object at 0x7eff517fd0a0>], comments=[]), isatools.model.DataFile(filename='EMseq_rawdata_DNA_sample_3_R1.fastq', label='Raw Data File R1', generated_from=[<isatools.model.Material object at 0x7eff517fd370>], comments=[]), isatools.model.DataFile(filename='EMseq_rawdata_DNA_sample_3_R2.fastq', label='Raw Data File R2', generated_from=[<isatools.model.Material object at 0x7eff517fd370>], comments=[])], samples=[isatools.model.Sample(name='sample_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[]), isatools.model.Sample(name='sample_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[]), isatools.model.Sample(name='sample_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[])], process_sequence=[isatools.model.Process(id=\"\". name=\"DNA_extraction_sample_1\", executes_protocol=Protocol(\n",
       "    name=DNA extraction\n",
       "    protocol_type=DNA extraction\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Sample(name='sample_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[])], outputs=[<isatools.model.Material object at 0x7eff946d2ac0>]), isatools.model.Process(id=\"\". name=\"WGS_DNA_sample_1\", executes_protocol=Protocol(\n",
       "    name=EM-seq\n",
       "    protocol_type=methylation profiling by high throughput sequencing\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7eff946d2ac0>], outputs=[isatools.model.DataFile(filename='EMseq_rawdata_DNA_sample_1_R1.fastq', label='Raw Data File R1', generated_from=[<isatools.model.Material object at 0x7eff946d2ac0>], comments=[]), isatools.model.DataFile(filename='EMseq_rawdata_DNA_sample_1_R2.fastq', label='Raw Data File R2', generated_from=[<isatools.model.Material object at 0x7eff946d2ac0>], comments=[])]), isatools.model.Process(id=\"\". name=\"DNA_extraction_sample_2\", executes_protocol=Protocol(\n",
       "    name=DNA extraction\n",
       "    protocol_type=DNA extraction\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Sample(name='sample_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[])], outputs=[<isatools.model.Material object at 0x7eff517fd0a0>]), isatools.model.Process(id=\"\". name=\"WGS_DNA_sample_2\", executes_protocol=Protocol(\n",
       "    name=EM-seq\n",
       "    protocol_type=methylation profiling by high throughput sequencing\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7eff517fd0a0>], outputs=[isatools.model.DataFile(filename='EMseq_rawdata_DNA_sample_2_R1.fastq', label='Raw Data File R1', generated_from=[<isatools.model.Material object at 0x7eff517fd0a0>], comments=[]), isatools.model.DataFile(filename='EMseq_rawdata_DNA_sample_2_R2.fastq', label='Raw Data File R2', generated_from=[<isatools.model.Material object at 0x7eff517fd0a0>], comments=[])]), isatools.model.Process(id=\"\". name=\"DNA_extraction_sample_3\", executes_protocol=Protocol(\n",
       "    name=DNA extraction\n",
       "    protocol_type=DNA extraction\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Sample(name='sample_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[])], outputs=[<isatools.model.Material object at 0x7eff517fd370>]), isatools.model.Process(id=\"\". name=\"WGS_DNA_sample_3\", executes_protocol=Protocol(\n",
       "    name=EM-seq\n",
       "    protocol_type=methylation profiling by high throughput sequencing\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7eff517fd370>], outputs=[isatools.model.DataFile(filename='EMseq_rawdata_DNA_sample_3_R1.fastq', label='Raw Data File R1', generated_from=[<isatools.model.Material object at 0x7eff517fd370>], comments=[]), isatools.model.DataFile(filename='EMseq_rawdata_DNA_sample_3_R2.fastq', label='Raw Data File R2', generated_from=[<isatools.model.Material object at 0x7eff517fd370>], comments=[])])], other_material=[<isatools.model.Material object at 0x7eff946d2ac0>, <isatools.model.Material object at 0x7eff517fd0a0>, <isatools.model.Material object at 0x7eff517fd370>], characteristic_categories=[], comments=[], units=[]), isatools.model.Assay(measurement_type=isatools.model.OntologyAnnotation(term='transcription profiling', term_source=isatools.model.OntologySource(name='EFO - Experimental Factor Ontology', file='http://www.ebi.ac.uk/efo/efo.owl', version='', description='The Experimental Factor Ontology (EFO) provides a systematic description of many experimental variables available in EBI databases, and for external projects such as the NHGRI GWAS catalogue. It combines parts of several biological ontologies, such as anatomy, disease and chemical compounds. The scope of EFO is to support the annotation, analysis and visualization of data handled by many groups at the EBI and as the core ontology for OpenTargets.org', comments=[]), term_accession='http://www.ebi.ac.uk/efo/EFO_0001032', comments=[]), technology_type=isatools.model.OntologyAnnotation(term='Nucleic Acid Sequencing', term_source=isatools.model.OntologySource(name='NCI Thesaurus OBO Edition', file='http://purl.obolibrary.org/obo/ncit.owl', version='', description='The NCIt OBO Edition project aims to increase integration of the NCIt with OBO Library ontologies. NCIt is a reference terminology that includes broad coverage of the cancer domain, including cancer related diseases, findings and abnormalities. NCIt OBO Edition releases should be considered experimental.', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCIT_C18881', comments=[]), technology_platform='OntologyAnnotation(\n",
       "    term=Illumina NovaSeq 6000\n",
       "    term_source=OBI - Ontology for Biomedical Investigations\n",
       "    term_accession=http://purl.obolibrary.org/obo/OBI_0002630\n",
       "    comments=0 Comment objects\n",
       ")', filename='a_assay_rnaseq_fimm.txt', data_files=[isatools.model.DataFile(filename='RNAseq_rawdata_RNA_sample_1_R1.fastq.gz', label='Raw Data File R1', generated_from=[<isatools.model.Material object at 0x7eff517fd700>], comments=[]), isatools.model.DataFile(filename='RNAseq_rawdata_RNA_sample_1_R2.fastq.gz', label='Raw Data File R2', generated_from=[<isatools.model.Material object at 0x7eff517fd700>], comments=[]), isatools.model.DataFile(filename='RNAseq_rawdata_RNA_sample_2_R1.fastq.gz', label='Raw Data File R1', generated_from=[<isatools.model.Material object at 0x7eff94715190>], comments=[]), isatools.model.DataFile(filename='RNAseq_rawdata_RNA_sample_2_R2.fastq.gz', label='Raw Data File R2', generated_from=[<isatools.model.Material object at 0x7eff94715190>], comments=[]), isatools.model.DataFile(filename='RNAseq_rawdata_RNA_sample_3_R1.fastq.gz', label='Raw Data File R1', generated_from=[<isatools.model.Material object at 0x7eff517fb670>], comments=[]), isatools.model.DataFile(filename='RNAseq_rawdata_RNA_sample_3_R2.fastq.gz', label='Raw Data File R2', generated_from=[<isatools.model.Material object at 0x7eff517fb670>], comments=[])], samples=[isatools.model.Sample(name='sample_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[]), isatools.model.Sample(name='sample_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[]), isatools.model.Sample(name='sample_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[])], process_sequence=[isatools.model.Process(id=\"\". name=\"RNA_extraction_sample_1\", executes_protocol=Protocol(\n",
       "    name=RNA extraction\n",
       "    protocol_type=RNA extraction\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Sample(name='sample_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[])], outputs=[<isatools.model.Material object at 0x7eff517fd700>]), isatools.model.Process(id=\"\". name=\"mRNAlibprep_RNA_sample_1\", executes_protocol=Protocol(\n",
       "    name=mRNA library preparation\n",
       "    protocol_type=library preparation protocol\n",
       "    uri=https://www.illumina.com/products/by-type/sequencing-kits/library-prep-kits/stranded-mrna-prep.html\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7eff517fd700>], outputs=[]), isatools.model.Process(id=\"\". name=\"mRNAseq_RNA_sample_1\", executes_protocol=Protocol(\n",
       "    name=mRNA Sequencing\n",
       "    protocol_type=nucleic acid sequencing\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[], outputs=[isatools.model.DataFile(filename='RNAseq_rawdata_RNA_sample_1_R1.fastq.gz', label='Raw Data File R1', generated_from=[<isatools.model.Material object at 0x7eff517fd700>], comments=[]), isatools.model.DataFile(filename='RNAseq_rawdata_RNA_sample_1_R2.fastq.gz', label='Raw Data File R2', generated_from=[<isatools.model.Material object at 0x7eff517fd700>], comments=[])]), isatools.model.Process(id=\"\". name=\"RNA_extraction_sample_2\", executes_protocol=Protocol(\n",
       "    name=RNA extraction\n",
       "    protocol_type=RNA extraction\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Sample(name='sample_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[])], outputs=[<isatools.model.Material object at 0x7eff94715190>]), isatools.model.Process(id=\"\". name=\"mRNAlibprep_RNA_sample_2\", executes_protocol=Protocol(\n",
       "    name=mRNA library preparation\n",
       "    protocol_type=library preparation protocol\n",
       "    uri=https://www.illumina.com/products/by-type/sequencing-kits/library-prep-kits/stranded-mrna-prep.html\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7eff94715190>], outputs=[]), isatools.model.Process(id=\"\". name=\"mRNAseq_RNA_sample_2\", executes_protocol=Protocol(\n",
       "    name=mRNA Sequencing\n",
       "    protocol_type=nucleic acid sequencing\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[], outputs=[isatools.model.DataFile(filename='RNAseq_rawdata_RNA_sample_2_R1.fastq.gz', label='Raw Data File R1', generated_from=[<isatools.model.Material object at 0x7eff94715190>], comments=[]), isatools.model.DataFile(filename='RNAseq_rawdata_RNA_sample_2_R2.fastq.gz', label='Raw Data File R2', generated_from=[<isatools.model.Material object at 0x7eff94715190>], comments=[])]), isatools.model.Process(id=\"\". name=\"RNA_extraction_sample_3\", executes_protocol=Protocol(\n",
       "    name=RNA extraction\n",
       "    protocol_type=RNA extraction\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Sample(name='sample_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[])], outputs=[<isatools.model.Material object at 0x7eff517fb670>]), isatools.model.Process(id=\"\". name=\"mRNAlibprep_RNA_sample_3\", executes_protocol=Protocol(\n",
       "    name=mRNA library preparation\n",
       "    protocol_type=library preparation protocol\n",
       "    uri=https://www.illumina.com/products/by-type/sequencing-kits/library-prep-kits/stranded-mrna-prep.html\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7eff517fb670>], outputs=[]), isatools.model.Process(id=\"\". name=\"mRNAseq_RNA_sample_3\", executes_protocol=Protocol(\n",
       "    name=mRNA Sequencing\n",
       "    protocol_type=nucleic acid sequencing\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[], outputs=[isatools.model.DataFile(filename='RNAseq_rawdata_RNA_sample_3_R1.fastq.gz', label='Raw Data File R1', generated_from=[<isatools.model.Material object at 0x7eff517fb670>], comments=[]), isatools.model.DataFile(filename='RNAseq_rawdata_RNA_sample_3_R2.fastq.gz', label='Raw Data File R2', generated_from=[<isatools.model.Material object at 0x7eff517fb670>], comments=[])])], other_material=[<isatools.model.Material object at 0x7eff517fd700>, <isatools.model.Material object at 0x7eff94715190>, <isatools.model.Material object at 0x7eff517fb670>], characteristic_categories=[], comments=[], units=[]), isatools.model.Assay(measurement_type=isatools.model.OntologyAnnotation(term='transcription profiling', term_source=isatools.model.OntologySource(name='EFO - Experimental Factor Ontology', file='http://www.ebi.ac.uk/efo/efo.owl', version='', description='The Experimental Factor Ontology (EFO) provides a systematic description of many experimental variables available in EBI databases, and for external projects such as the NHGRI GWAS catalogue. It combines parts of several biological ontologies, such as anatomy, disease and chemical compounds. The scope of EFO is to support the annotation, analysis and visualization of data handled by many groups at the EBI and as the core ontology for OpenTargets.org', comments=[]), term_accession='http://www.ebi.ac.uk/efo/EFO_0001032', comments=[]), technology_type=isatools.model.OntologyAnnotation(term='microRNA profiling by RT-PCR', term_source=isatools.model.OntologySource(name='EFO - Experimental Factor Ontology', file='http://www.ebi.ac.uk/efo/efo.owl', version='', description='The Experimental Factor Ontology (EFO) provides a systematic description of many experimental variables available in EBI databases, and for external projects such as the NHGRI GWAS catalogue. It combines parts of several biological ontologies, such as anatomy, disease and chemical compounds. The scope of EFO is to support the annotation, analysis and visualization of data handled by many groups at the EBI and as the core ontology for OpenTargets.org', comments=[]), term_accession='http://www.ebi.ac.uk/efo/EFO_0007687', comments=[]), technology_platform='OntologyAnnotation(\n",
       "    term=LightCycler 480 Real-Time PCR detection instrument\n",
       "    term_source=BAO - BioAssay Ontology\n",
       "    term_accession=http://www.bioassayontology.org/bao#BAO_0150063\n",
       "    comments=0 Comment objects\n",
       ")', filename='a_assay_mirnaqrtpcr_sermas.txt', data_files=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_1_Replicate_1', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff51811b20>], comments=[]), isatools.model.DataFile(filename='miRNAqPCR_normalizationdata_sample_1_Replicate_1', label='Derived Data File', generated_from=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_1_Replicate_1', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff51811b20>], comments=[])], comments=[]), isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_1_Replicate_2', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff51811b20>], comments=[]), isatools.model.DataFile(filename='miRNAqPCR_normalizationdata_sample_1_Replicate_2', label='Derived Data File', generated_from=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_1_Replicate_2', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff51811b20>], comments=[])], comments=[]), isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_1_Replicate_3', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff51811b20>], comments=[]), isatools.model.DataFile(filename='miRNAqPCR_normalizationdata_sample_1_Replicate_3', label='Derived Data File', generated_from=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_1_Replicate_3', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff51811b20>], comments=[])], comments=[]), isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_2_Replicate_1', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff517f4100>], comments=[]), isatools.model.DataFile(filename='miRNAqPCR_normalizationdata_sample_2_Replicate_1', label='Derived Data File', generated_from=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_2_Replicate_1', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff517f4100>], comments=[])], comments=[]), isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_2_Replicate_2', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff517f4100>], comments=[]), isatools.model.DataFile(filename='miRNAqPCR_normalizationdata_sample_2_Replicate_2', label='Derived Data File', generated_from=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_2_Replicate_2', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff517f4100>], comments=[])], comments=[]), isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_2_Replicate_3', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff517f4100>], comments=[]), isatools.model.DataFile(filename='miRNAqPCR_normalizationdata_sample_2_Replicate_3', label='Derived Data File', generated_from=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_2_Replicate_3', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff517f4100>], comments=[])], comments=[]), isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_3_Replicate_1', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff517f4670>], comments=[]), isatools.model.DataFile(filename='miRNAqPCR_normalizationdata_sample_3_Replicate_1', label='Derived Data File', generated_from=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_3_Replicate_1', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff517f4670>], comments=[])], comments=[]), isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_3_Replicate_2', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff517f4670>], comments=[]), isatools.model.DataFile(filename='miRNAqPCR_normalizationdata_sample_3_Replicate_2', label='Derived Data File', generated_from=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_3_Replicate_2', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff517f4670>], comments=[])], comments=[]), isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_3_Replicate_3', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff517f4670>], comments=[]), isatools.model.DataFile(filename='miRNAqPCR_normalizationdata_sample_3_Replicate_3', label='Derived Data File', generated_from=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_3_Replicate_3', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff517f4670>], comments=[])], comments=[])], samples=[isatools.model.Sample(name='sample_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[]), isatools.model.Sample(name='sample_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[]), isatools.model.Sample(name='sample_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[]), isatools.model.Sample(name='sample_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[]), isatools.model.Sample(name='sample_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[]), isatools.model.Sample(name='sample_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[]), isatools.model.Sample(name='sample_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[]), isatools.model.Sample(name='sample_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[]), isatools.model.Sample(name='sample_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[])], process_sequence=[isatools.model.Process(id=\"\". name=\"RNA_extraction_sample_1\", executes_protocol=Protocol(\n",
       "    name=RNA extraction\n",
       "    protocol_type=RNA extraction\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Sample(name='sample_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[])], outputs=[<isatools.model.Material object at 0x7eff51811a90>]), isatools.model.Process(id=\"\". name=\"ReverseTranscription_RNA_sample_1\", executes_protocol=Protocol(\n",
       "    name=Reverse Transcription\n",
       "    protocol_type=artificially induced reverse transcription\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7eff51811a90>], outputs=[<isatools.model.Material object at 0x7eff51811b20>]), isatools.model.Process(id=\"\". name=\"miRNAarray_sample_1_Replicate_1\", executes_protocol=Protocol(\n",
       "    name=miRNA array\n",
       "    protocol_type=microRNA profiling by RT-PCR\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=1 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7eff51811b20>], outputs=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_1_Replicate_1', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff51811b20>], comments=[])]), isatools.model.Process(id=\"\". name=\"datanormalization_sample_1_Replicate_1\", executes_protocol=Protocol(\n",
       "    name=Data Normalization\n",
       "    protocol_type=Constant Normalization\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_1_Replicate_1', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff51811b20>], comments=[])], outputs=[isatools.model.DataFile(filename='miRNAqPCR_normalizationdata_sample_1_Replicate_1', label='Derived Data File', generated_from=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_1_Replicate_1', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff51811b20>], comments=[])], comments=[])]), isatools.model.Process(id=\"\". name=\"RNA_extraction_sample_1\", executes_protocol=Protocol(\n",
       "    name=RNA extraction\n",
       "    protocol_type=RNA extraction\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Sample(name='sample_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[])], outputs=[<isatools.model.Material object at 0x7eff51811a90>]), isatools.model.Process(id=\"\". name=\"ReverseTranscription_RNA_sample_1\", executes_protocol=Protocol(\n",
       "    name=Reverse Transcription\n",
       "    protocol_type=artificially induced reverse transcription\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7eff51811a90>], outputs=[<isatools.model.Material object at 0x7eff51811b20>]), isatools.model.Process(id=\"\". name=\"miRNAarray_sample_1_Replicate_2\", executes_protocol=Protocol(\n",
       "    name=miRNA array\n",
       "    protocol_type=microRNA profiling by RT-PCR\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=1 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7eff51811b20>], outputs=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_1_Replicate_2', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff51811b20>], comments=[])]), isatools.model.Process(id=\"\". name=\"datanormalization_sample_1_Replicate_2\", executes_protocol=Protocol(\n",
       "    name=Data Normalization\n",
       "    protocol_type=Constant Normalization\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_1_Replicate_2', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff51811b20>], comments=[])], outputs=[isatools.model.DataFile(filename='miRNAqPCR_normalizationdata_sample_1_Replicate_2', label='Derived Data File', generated_from=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_1_Replicate_2', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff51811b20>], comments=[])], comments=[])]), isatools.model.Process(id=\"\". name=\"RNA_extraction_sample_1\", executes_protocol=Protocol(\n",
       "    name=RNA extraction\n",
       "    protocol_type=RNA extraction\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Sample(name='sample_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[])], outputs=[<isatools.model.Material object at 0x7eff51811a90>]), isatools.model.Process(id=\"\". name=\"ReverseTranscription_RNA_sample_1\", executes_protocol=Protocol(\n",
       "    name=Reverse Transcription\n",
       "    protocol_type=artificially induced reverse transcription\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7eff51811a90>], outputs=[<isatools.model.Material object at 0x7eff51811b20>]), isatools.model.Process(id=\"\". name=\"miRNAarray_sample_1_Replicate_3\", executes_protocol=Protocol(\n",
       "    name=miRNA array\n",
       "    protocol_type=microRNA profiling by RT-PCR\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=1 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7eff51811b20>], outputs=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_1_Replicate_3', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff51811b20>], comments=[])]), isatools.model.Process(id=\"\". name=\"datanormalization_sample_1_Replicate_3\", executes_protocol=Protocol(\n",
       "    name=Data Normalization\n",
       "    protocol_type=Constant Normalization\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_1_Replicate_3', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff51811b20>], comments=[])], outputs=[isatools.model.DataFile(filename='miRNAqPCR_normalizationdata_sample_1_Replicate_3', label='Derived Data File', generated_from=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_1_Replicate_3', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff51811b20>], comments=[])], comments=[])]), isatools.model.Process(id=\"\". name=\"RNA_extraction_sample_2\", executes_protocol=Protocol(\n",
       "    name=RNA extraction\n",
       "    protocol_type=RNA extraction\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Sample(name='sample_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[])], outputs=[<isatools.model.Material object at 0x7eff51811be0>]), isatools.model.Process(id=\"\". name=\"ReverseTranscription_RNA_sample_2\", executes_protocol=Protocol(\n",
       "    name=Reverse Transcription\n",
       "    protocol_type=artificially induced reverse transcription\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7eff51811be0>], outputs=[<isatools.model.Material object at 0x7eff517f4100>]), isatools.model.Process(id=\"\". name=\"miRNAarray_sample_2_Replicate_1\", executes_protocol=Protocol(\n",
       "    name=miRNA array\n",
       "    protocol_type=microRNA profiling by RT-PCR\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=1 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7eff517f4100>], outputs=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_2_Replicate_1', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff517f4100>], comments=[])]), isatools.model.Process(id=\"\". name=\"datanormalization_sample_2_Replicate_1\", executes_protocol=Protocol(\n",
       "    name=Data Normalization\n",
       "    protocol_type=Constant Normalization\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_2_Replicate_1', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff517f4100>], comments=[])], outputs=[isatools.model.DataFile(filename='miRNAqPCR_normalizationdata_sample_2_Replicate_1', label='Derived Data File', generated_from=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_2_Replicate_1', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff517f4100>], comments=[])], comments=[])]), isatools.model.Process(id=\"\". name=\"RNA_extraction_sample_2\", executes_protocol=Protocol(\n",
       "    name=RNA extraction\n",
       "    protocol_type=RNA extraction\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Sample(name='sample_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[])], outputs=[<isatools.model.Material object at 0x7eff51811be0>]), isatools.model.Process(id=\"\". name=\"ReverseTranscription_RNA_sample_2\", executes_protocol=Protocol(\n",
       "    name=Reverse Transcription\n",
       "    protocol_type=artificially induced reverse transcription\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7eff51811be0>], outputs=[<isatools.model.Material object at 0x7eff517f4100>]), isatools.model.Process(id=\"\". name=\"miRNAarray_sample_2_Replicate_2\", executes_protocol=Protocol(\n",
       "    name=miRNA array\n",
       "    protocol_type=microRNA profiling by RT-PCR\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=1 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7eff517f4100>], outputs=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_2_Replicate_2', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff517f4100>], comments=[])]), isatools.model.Process(id=\"\". name=\"datanormalization_sample_2_Replicate_2\", executes_protocol=Protocol(\n",
       "    name=Data Normalization\n",
       "    protocol_type=Constant Normalization\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_2_Replicate_2', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff517f4100>], comments=[])], outputs=[isatools.model.DataFile(filename='miRNAqPCR_normalizationdata_sample_2_Replicate_2', label='Derived Data File', generated_from=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_2_Replicate_2', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff517f4100>], comments=[])], comments=[])]), isatools.model.Process(id=\"\". name=\"RNA_extraction_sample_2\", executes_protocol=Protocol(\n",
       "    name=RNA extraction\n",
       "    protocol_type=RNA extraction\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Sample(name='sample_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[])], outputs=[<isatools.model.Material object at 0x7eff51811be0>]), isatools.model.Process(id=\"\". name=\"ReverseTranscription_RNA_sample_2\", executes_protocol=Protocol(\n",
       "    name=Reverse Transcription\n",
       "    protocol_type=artificially induced reverse transcription\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7eff51811be0>], outputs=[<isatools.model.Material object at 0x7eff517f4100>]), isatools.model.Process(id=\"\". name=\"miRNAarray_sample_2_Replicate_3\", executes_protocol=Protocol(\n",
       "    name=miRNA array\n",
       "    protocol_type=microRNA profiling by RT-PCR\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=1 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7eff517f4100>], outputs=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_2_Replicate_3', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff517f4100>], comments=[])]), isatools.model.Process(id=\"\". name=\"datanormalization_sample_2_Replicate_3\", executes_protocol=Protocol(\n",
       "    name=Data Normalization\n",
       "    protocol_type=Constant Normalization\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_2_Replicate_3', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff517f4100>], comments=[])], outputs=[isatools.model.DataFile(filename='miRNAqPCR_normalizationdata_sample_2_Replicate_3', label='Derived Data File', generated_from=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_2_Replicate_3', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff517f4100>], comments=[])], comments=[])]), isatools.model.Process(id=\"\". name=\"RNA_extraction_sample_3\", executes_protocol=Protocol(\n",
       "    name=RNA extraction\n",
       "    protocol_type=RNA extraction\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Sample(name='sample_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[])], outputs=[<isatools.model.Material object at 0x7eff517f41c0>]), isatools.model.Process(id=\"\". name=\"ReverseTranscription_RNA_sample_3\", executes_protocol=Protocol(\n",
       "    name=Reverse Transcription\n",
       "    protocol_type=artificially induced reverse transcription\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7eff517f41c0>], outputs=[<isatools.model.Material object at 0x7eff517f4670>]), isatools.model.Process(id=\"\". name=\"miRNAarray_sample_3_Replicate_1\", executes_protocol=Protocol(\n",
       "    name=miRNA array\n",
       "    protocol_type=microRNA profiling by RT-PCR\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=1 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7eff517f4670>], outputs=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_3_Replicate_1', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff517f4670>], comments=[])]), isatools.model.Process(id=\"\". name=\"datanormalization_sample_3_Replicate_1\", executes_protocol=Protocol(\n",
       "    name=Data Normalization\n",
       "    protocol_type=Constant Normalization\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_3_Replicate_1', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff517f4670>], comments=[])], outputs=[isatools.model.DataFile(filename='miRNAqPCR_normalizationdata_sample_3_Replicate_1', label='Derived Data File', generated_from=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_3_Replicate_1', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff517f4670>], comments=[])], comments=[])]), isatools.model.Process(id=\"\". name=\"RNA_extraction_sample_3\", executes_protocol=Protocol(\n",
       "    name=RNA extraction\n",
       "    protocol_type=RNA extraction\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Sample(name='sample_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[])], outputs=[<isatools.model.Material object at 0x7eff517f41c0>]), isatools.model.Process(id=\"\". name=\"ReverseTranscription_RNA_sample_3\", executes_protocol=Protocol(\n",
       "    name=Reverse Transcription\n",
       "    protocol_type=artificially induced reverse transcription\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7eff517f41c0>], outputs=[<isatools.model.Material object at 0x7eff517f4670>]), isatools.model.Process(id=\"\". name=\"miRNAarray_sample_3_Replicate_2\", executes_protocol=Protocol(\n",
       "    name=miRNA array\n",
       "    protocol_type=microRNA profiling by RT-PCR\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=1 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7eff517f4670>], outputs=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_3_Replicate_2', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff517f4670>], comments=[])]), isatools.model.Process(id=\"\". name=\"datanormalization_sample_3_Replicate_2\", executes_protocol=Protocol(\n",
       "    name=Data Normalization\n",
       "    protocol_type=Constant Normalization\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_3_Replicate_2', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff517f4670>], comments=[])], outputs=[isatools.model.DataFile(filename='miRNAqPCR_normalizationdata_sample_3_Replicate_2', label='Derived Data File', generated_from=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_3_Replicate_2', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff517f4670>], comments=[])], comments=[])]), isatools.model.Process(id=\"\". name=\"RNA_extraction_sample_3\", executes_protocol=Protocol(\n",
       "    name=RNA extraction\n",
       "    protocol_type=RNA extraction\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Sample(name='sample_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[])], outputs=[<isatools.model.Material object at 0x7eff517f41c0>]), isatools.model.Process(id=\"\". name=\"ReverseTranscription_RNA_sample_3\", executes_protocol=Protocol(\n",
       "    name=Reverse Transcription\n",
       "    protocol_type=artificially induced reverse transcription\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7eff517f41c0>], outputs=[<isatools.model.Material object at 0x7eff517f4670>]), isatools.model.Process(id=\"\". name=\"miRNAarray_sample_3_Replicate_3\", executes_protocol=Protocol(\n",
       "    name=miRNA array\n",
       "    protocol_type=microRNA profiling by RT-PCR\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=1 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7eff517f4670>], outputs=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_3_Replicate_3', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff517f4670>], comments=[])]), isatools.model.Process(id=\"\". name=\"datanormalization_sample_3_Replicate_3\", executes_protocol=Protocol(\n",
       "    name=Data Normalization\n",
       "    protocol_type=Constant Normalization\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_3_Replicate_3', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff517f4670>], comments=[])], outputs=[isatools.model.DataFile(filename='miRNAqPCR_normalizationdata_sample_3_Replicate_3', label='Derived Data File', generated_from=[isatools.model.DataFile(filename='miRNAqPCR_rawdata_sample_3_Replicate_3', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7eff517f4670>], comments=[])], comments=[])])], other_material=[<isatools.model.Material object at 0x7eff51811a90>, <isatools.model.Material object at 0x7eff51811b20>, <isatools.model.Material object at 0x7eff51811a90>, <isatools.model.Material object at 0x7eff51811b20>, <isatools.model.Material object at 0x7eff51811a90>, <isatools.model.Material object at 0x7eff51811b20>, <isatools.model.Material object at 0x7eff51811be0>, <isatools.model.Material object at 0x7eff517f4100>, <isatools.model.Material object at 0x7eff51811be0>, <isatools.model.Material object at 0x7eff517f4100>, <isatools.model.Material object at 0x7eff51811be0>, <isatools.model.Material object at 0x7eff517f4100>, <isatools.model.Material object at 0x7eff517f41c0>, <isatools.model.Material object at 0x7eff517f4670>, <isatools.model.Material object at 0x7eff517f41c0>, <isatools.model.Material object at 0x7eff517f4670>, <isatools.model.Material object at 0x7eff517f41c0>, <isatools.model.Material object at 0x7eff517f4670>], characteristic_categories=[], comments=[], units=[]), isatools.model.Assay(measurement_type=isatools.model.OntologyAnnotation(term='targeted metabolite profiling', term_source=isatools.model.OntologySource(name='MS - Mass spectrometry ontology', file='http://purl.obolibrary.org/obo/msio.owl', version='', description='MSIO aims to provide a single point of entry to support semantic markup of experiments making use of NMR and MS techniques to identify, measure and quantify small molecules known as metabolites. MSIO covers metabolite profiling, targeted or undertargeted, tracer based applications. MSIO reuses a number of resources such as CHEBI, DUO, NMRCV, OBI, and STATO.', comments=[]), term_accession='http://purl.obolibrary.org/obo/MSIO_0000100', comments=[]), technology_type=isatools.model.OntologyAnnotation(term='mass spectrometry', term_source=isatools.model.OntologySource(name='CHMO - Chemical Methods Ontology', file='http://purl.obolibrary.org/obo/chmo.owl', version='', description='CHMO, the chemical methods ontology, describes methods used to collect data in chemical experiments, such as mass spectrometry and electron microscopy prepare and separate material for further analysis, such as sample ionisation, chromatography, and electrophoresis synthesise materials, such as epitaxy and continuous vapour deposition It also describes the instruments used in these experiments, such as mass spectrometers and chromatography columns. It is intended to be complementary to the Ontology for Biomedical Investigations (OBI).', comments=[]), term_accession='http://purl.obolibrary.org/obo/CHMO_0000470', comments=[]), technology_platform='Micromass Quattro II triple-quadrupole mass spectrometer', filename='a_assay_metabolomics_acylcarnitines_rumc.txt', data_files=[], samples=[isatools.model.Sample(name='sample_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[]), isatools.model.Sample(name='sample_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[]), isatools.model.Sample(name='sample_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[])], process_sequence=[isatools.model.Process(id=\"\". name=\"Sample_preparation_plasma_acylcarnitine_analysis_sample_1\", executes_protocol=Protocol(\n",
       "    name=Extraction\n",
       "    protocol_type=extraction\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=2 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Sample(name='sample_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[])], outputs=[<isatools.model.Material object at 0x7eff518023d0>]), isatools.model.Process(id=\"\". name=\"Chromatography_plasma_acylcarnitine_analysis_sample_1\", executes_protocol=Protocol(\n",
       "    name=Chromatography\n",
       "    protocol_type=chromatography\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=3 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7eff518023d0>], outputs=[]), isatools.model.Process(id=\"\". name=\"MS_plasma_acylcarnitine_analysis_sample_1\", executes_protocol=Protocol(\n",
       "    name=Mass spectrometry\n",
       "    protocol_type=mass spectrometry\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=6 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[], outputs=[isatools.model.DataFile(filename='sample_1^B-BCARP_NM.raw', label='Raw Spectral Data File', generated_from=[], comments=[])]), isatools.model.Process(id=\"\". name=\"Data_transformation_plasma_acylcarnitine_analysis_sample_1\", executes_protocol=Protocol(\n",
       "    name=Data transformation - Plasma acylcarnitine analysis\n",
       "    protocol_type=data transformation\n",
       "    uri=https://doi.org/10.1023/A:1005587617745\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.DataFile(filename='sample_1^B-BCARP_NM.raw', label='Raw Spectral Data File', generated_from=[], comments=[])], outputs=[]), isatools.model.Process(id=\"\". name=\"Metabolite_identification_plasma_acylcarnitine_analysis_sample_1\", executes_protocol=Protocol(\n",
       "    name=Metabolite identification - Plasma acylcarnitine analysis\n",
       "    protocol_type=metabolite identification\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[], outputs=[isatools.model.DataFile(filename='m_carnprofiel.tsv', label='Metabolite Assignment File', generated_from=[], comments=[])]), isatools.model.Process(id=\"\". name=\"Sample_preparation_plasma_acylcarnitine_analysis_sample_2\", executes_protocol=Protocol(\n",
       "    name=Extraction\n",
       "    protocol_type=extraction\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=2 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Sample(name='sample_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[])], outputs=[<isatools.model.Material object at 0x7eff51802a90>]), isatools.model.Process(id=\"\". name=\"Chromatography_plasma_acylcarnitine_analysis_sample_2\", executes_protocol=Protocol(\n",
       "    name=Chromatography\n",
       "    protocol_type=chromatography\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=3 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7eff51802a90>], outputs=[]), isatools.model.Process(id=\"\". name=\"MS_plasma_acylcarnitine_analysis_sample_2\", executes_protocol=Protocol(\n",
       "    name=Mass spectrometry\n",
       "    protocol_type=mass spectrometry\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=6 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[], outputs=[isatools.model.DataFile(filename='sample_2^B-BCARP_NM.raw', label='Raw Spectral Data File', generated_from=[], comments=[])]), isatools.model.Process(id=\"\". name=\"Data_transformation_plasma_acylcarnitine_analysis_sample_2\", executes_protocol=Protocol(\n",
       "    name=Data transformation - Plasma acylcarnitine analysis\n",
       "    protocol_type=data transformation\n",
       "    uri=https://doi.org/10.1023/A:1005587617745\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.DataFile(filename='sample_2^B-BCARP_NM.raw', label='Raw Spectral Data File', generated_from=[], comments=[])], outputs=[]), isatools.model.Process(id=\"\". name=\"Metabolite_identification_plasma_acylcarnitine_analysis_sample_2\", executes_protocol=Protocol(\n",
       "    name=Metabolite identification - Plasma acylcarnitine analysis\n",
       "    protocol_type=metabolite identification\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[], outputs=[isatools.model.DataFile(filename='m_carnprofiel.tsv', label='Metabolite Assignment File', generated_from=[], comments=[])]), isatools.model.Process(id=\"\". name=\"Sample_preparation_plasma_acylcarnitine_analysis_sample_3\", executes_protocol=Protocol(\n",
       "    name=Extraction\n",
       "    protocol_type=extraction\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=2 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Sample(name='sample_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[])], outputs=[<isatools.model.Material object at 0x7eff51826130>]), isatools.model.Process(id=\"\". name=\"Chromatography_plasma_acylcarnitine_analysis_sample_3\", executes_protocol=Protocol(\n",
       "    name=Chromatography\n",
       "    protocol_type=chromatography\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=3 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7eff51826130>], outputs=[]), isatools.model.Process(id=\"\". name=\"MS_plasma_acylcarnitine_analysis_sample_3\", executes_protocol=Protocol(\n",
       "    name=Mass spectrometry\n",
       "    protocol_type=mass spectrometry\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=6 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[], outputs=[isatools.model.DataFile(filename='sample_3^B-BCARP_NM.raw', label='Raw Spectral Data File', generated_from=[], comments=[])]), isatools.model.Process(id=\"\". name=\"Data_transformation_plasma_acylcarnitine_analysis_sample_3\", executes_protocol=Protocol(\n",
       "    name=Data transformation - Plasma acylcarnitine analysis\n",
       "    protocol_type=data transformation\n",
       "    uri=https://doi.org/10.1023/A:1005587617745\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.DataFile(filename='sample_3^B-BCARP_NM.raw', label='Raw Spectral Data File', generated_from=[], comments=[])], outputs=[]), isatools.model.Process(id=\"\". name=\"Metabolite_identification_plasma_acylcarnitine_analysis_sample_3\", executes_protocol=Protocol(\n",
       "    name=Metabolite identification - Plasma acylcarnitine analysis\n",
       "    protocol_type=metabolite identification\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[], outputs=[isatools.model.DataFile(filename='m_carnprofiel.tsv', label='Metabolite Assignment File', generated_from=[], comments=[])])], other_material=[<isatools.model.Material object at 0x7eff518023d0>, <isatools.model.Material object at 0x7eff51802a90>, <isatools.model.Material object at 0x7eff51826130>], characteristic_categories=[], comments=[], units=[])], sources=[isatools.model.Source(name='individual_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[]), isatools.model.Source(name='individual_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[]), isatools.model.Source(name='individual_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], samples=[isatools.model.Sample(name='sample_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[]), isatools.model.Sample(name='sample_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[]), isatools.model.Sample(name='sample_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[])], process_sequence=[isatools.model.Process(id=\"\". name=\"samplecollection_1\", executes_protocol=Protocol(\n",
       "    name=sample_collection_protocol\n",
       "    protocol_type=sample collection\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Source(name='individual_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], outputs=[isatools.model.Sample(name='sample_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[])]), isatools.model.Process(id=\"\". name=\"samplecollection_2\", executes_protocol=Protocol(\n",
       "    name=sample_collection_protocol\n",
       "    protocol_type=sample collection\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Source(name='individual_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], outputs=[isatools.model.Sample(name='sample_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[])]), isatools.model.Process(id=\"\". name=\"samplecollection_3\", executes_protocol=Protocol(\n",
       "    name=sample_collection_protocol\n",
       "    protocol_type=sample collection\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Source(name='individual_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], outputs=[isatools.model.Sample(name='sample_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), value=isatools.model.OntologyAnnotation(term='blood', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0000178', comments=[]), unit=None, comments=[])], factor_values=[], derives_from=[isatools.model.Source(name='individual_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], comments=[])])], other_material=[], characteristic_categories=[], comments=[], units=[])], comments=[])"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# create output directories (if necessary)\n",
    "import os\n",
    "out_dir = \"../ISA\"\n",
    "out_dir_tab = os.path.join(out_dir, \"ISA-Tab\")\n",
    "out_dir_tabxlsx = os.path.join(out_dir, \"ISA-Tab_asXLSX\")\n",
    "out_dir_json = os.path.join(out_dir, \"ISA-JSON\")\n",
    "for p in (out_dir_tab, out_dir_tabxlsx, out_dir_json):\n",
    "    if not os.path.isdir(p):\n",
    "        os.makedirs(p)\n",
    "# write to ISA-Tab\n",
    "from isatools import isatab\n",
    "isatab.dump(investigation, out_dir_tab)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read ISA-Tab files and validate\n",
    "#with open(os.path.join(out_dir, \"i_investigation.txt\")) as my_file:\n",
    "#    ISA = isatab.validate(my_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Write ISA-JSON file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "# write to ISA-JSON\n",
    "# see example: https://isa-tools.org/isa-api/content/examples/example-createSimpleISAJSON.html\n",
    "import json\n",
    "from isatools.isajson import ISAJSONEncoder\n",
    "with open(os.path.join(out_dir_json, \"isa.json\"), \"w\") as out_file:\n",
    "    json.dump(\n",
    "        investigation, \n",
    "        out_file,\n",
    "        cls = ISAJSONEncoder, \n",
    "        sort_keys = True, \n",
    "        indent = 4, \n",
    "        separators = (',', ': '))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#from isatools.convert import json2isatab\n",
    "#from isatools import isajson\n",
    "#isajson.validate(open('isa.json'))\n",
    "#with open(\"isa.json\") as file_pointer:\n",
    "#    json2isatab.convert(file_pointer, './ISA/')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Combine csv files to xlsx file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "import csv\n",
    "import openpyxl\n",
    "\n",
    "xlsx_workbook = openpyxl.Workbook()\n",
    "for idx, f in enumerate([investigation_filename, study_filename] + \n",
    "                        assay_filenames):\n",
    "    xlsx_sheets = xlsx_workbook.create_sheet(f[:30], idx)\n",
    "    with open(os.path.join(out_dir_tab, f), \"r\") as txt_file:\n",
    "        reader = csv.reader(txt_file, delimiter = \"\\t\")\n",
    "        for row in reader:\n",
    "            xlsx_sheets.append(row)\n",
    "xlsx_workbook.save(os.path.join(out_dir_tabxlsx, \"ISA_merged.xlsx\"))"
   ]
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "8dc67bcfc54321ed9774b432913c7e6eb9fda909d4295e73f8e0cfce319139b9"
  },
  "kernelspec": {
   "display_name": "Python 3.9.6 ('python3isa')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
