{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create ISA objects for EATRIS-Plus multi-omics data set of Czech population cohort\n",
    "\n",
    "See example: https://isatools.readthedocs.io/en/latest/example-createSimpleISAtab.html"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Define ontology references"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from isatools.model import *\n",
    "ontologies = {\n",
    "    \"CHEBI\": OntologySource(\n",
    "        name = \"CHEBI - Chemical Entities of Biological Interest\", \n",
    "        file = \"http://purl.obolibrary.org/obo/chebi.owl\",\n",
    "        description = \"A structured classification of molecular entities of biological interest focusing on 'small' chemical compounds.\"), \n",
    "    \"CRO\": OntologySource(\n",
    "        # The Contributor Role Ontology (CRO) is an extension of the CASRAI Contributor Roles Taxonomy (CRediT) and replaces the former Contribution Ontology.\n",
    "        name = \"CRO - Contributor Role Ontology\",\n",
    "        file = \"http://purl.obolibrary.org/obo/cro.owl\",\n",
    "        description = \"A classification of the diverse roles performed in the work leading to a published research output in the sciences. Its purpose to provide transparency in contributions to scholarly published work, to enable improved systems of attribution, credit, and accountability.\"),\n",
    "    \"EDAM\": OntologySource(\n",
    "        name = \"EDAM - EMBRACE Data and Methods\", \n",
    "        file = \"http://edamontology.org/EDAM.owl\",\n",
    "        description = \"EDAM (EMBRACE Data and Methods) is an ontology of common bioinformatics operations, topics, types of data including identifiers, and formats. EDAM comprises common concepts (shared within the bioinformatics community) that apply to semantic annotation of resources.\"), \n",
    "    \"EFO\": OntologySource(\n",
    "        name = \"EFO - Experimental Factor Ontology\", \n",
    "        file = \"http://www.ebi.ac.uk/efo/efo.owl\",\n",
    "        description = \"The Experimental Factor Ontology (EFO) provides a systematic description of many experimental variables available in EBI databases, and for external projects such as the NHGRI GWAS catalogue. It combines parts of several biological ontologies, such as anatomy, disease and chemical compounds. The scope of EFO is to support the annotation, analysis and visualization of data handled by many groups at the EBI and as the core ontology for OpenTargets.org\"), \n",
    "    \"GENEPIO\": OntologySource(\n",
    "        name = \"GENEPIO - Genomic Epidemiology Ontology\", \n",
    "        file = \"http://purl.obolibrary.org/obo/genepio.owl\",\n",
    "        description = \"The Genomic Epidemiology Ontology (GenEpiO) covers vocabulary necessary to identify, document and research foodborne pathogens and associated outbreaks.\"),\n",
    "    \"MMO\": OntologySource(\n",
    "        name = \"MMO - Measurement method ontology\", \n",
    "        file = \"http://purl.obolibrary.org/obo/mmo.owl\",\n",
    "        description = \"A representation of the variety of methods used to make clinical and phenotype measurements.\"),\n",
    "    \"NCBITAXON\": OntologySource(\n",
    "        name = \"NCBI organismal classification\", \n",
    "        file = \"http://purl.obolibrary.org/obo/ncbitaxon.owl\",\n",
    "        description = \"An ontology representation of the NCBI organismal taxonomy\"),\n",
    "    \"NCIT\": OntologySource(\n",
    "        name = \"NCI Thesaurus OBO Edition\", \n",
    "        file = \"http://purl.obolibrary.org/obo/ncit.owl\",\n",
    "        description = \"The NCIt OBO Edition project aims to increase integration of the NCIt with OBO Library ontologies. NCIt is a reference terminology that includes broad coverage of the cancer domain, including cancer related diseases, findings and abnormalities. NCIt OBO Edition releases should be considered experimental.\"),\n",
    "    \"OBI\": OntologySource(\n",
    "        name = \"OBI - Ontology for Biomedical Investigations\", \n",
    "        file = \"http://purl.obolibrary.org/obo/obi.owl\",\n",
    "        description = \"An integrated ontology for the description of life-science and clinical investigations\"),\n",
    "    \"OMIABIS\": OntologySource(\n",
    "        name = \"Ontologized MIABIS\", \n",
    "        file = \"http://purl.obolibrary.org/obo/omiabis.owl\",\n",
    "        description = \"An ontological version of MIABIS (Minimum Information About BIobank data Sharing)\"),\n",
    "    \"PRIDE\": OntologySource(\n",
    "        name = \"PRIDE Controlled Vocabulary\",\n",
    "        file = \"http://purl.obolibrary.org/obo/pride_cv.obo\",\n",
    "        description = \"The PRIDE PRoteomics IDEntifications (PRIDE) database is a centralized, standards compliant, public data repository for proteomics data, including protein and peptide identifications, post-translational modifications and supporting spectral evidence.\"),\n",
    "    \"STATO\": OntologySource(\n",
    "        name = \"STATO: the statistical methods ontology\",\n",
    "        file = \"http://purl.obolibrary.org/obo/stato.owl\",\n",
    "        description = \"STATO is the statistical methods ontology. It contains concepts and properties related to statistical methods, probability distributions and other concepts related to statistical analysis, including relationships to study designs and plots.\"),\n",
    "    \"UBERON\": OntologySource(\n",
    "        name = \"Uber-anatomy ontology\",\n",
    "        file = \"http://purl.obolibrary.org/obo/uberon.owl\",\n",
    "        description = \"Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.\")\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create investigation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "investigation = Investigation(\n",
    "    filename = \"i_investigation.txt\", \n",
    "    identifier = \"\", \n",
    "    title = \"EATRIS-Plus - Flagship in Personalised Medicine\",\n",
    "    description = \"EATRIS-Plus project aims to support the long-term sustainability of the European Research Infrastructure for Translational Medicine (EATRIS) by delivering innovative scientific tools to the research community, strengthening the EATRIS financial model, and reinforcing EATRIS’ leadership in the European Research Area in the field of Personalised Medicine (PM).\",\n",
    "    submission_date = \"\",\n",
    "    public_release_date = \"\",\n",
    "    ontology_source_references = [o for o in ontologies.values()],\n",
    "    publications = None,\n",
    "    contacts = [\n",
    "        Person(\n",
    "            last_name = \"Keidong\", \n",
    "            first_name = \"Eliis\",\n",
    "            #mid_initials = \"\",\n",
    "            affiliation = \"EATRIS\",\n",
    "            roles = [\n",
    "                OntologyAnnotation(\n",
    "                    term = \"project management role\",\n",
    "                    term_source = ontologies[\"CRO\"], \n",
    "                    term_accession =\"http://purl.obolibrary.org/obo/CRO_0000065\")])],\n",
    "    studies = None,\n",
    "    comments = None)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create study"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "cohort_study = Study(\n",
    "    filename = \"s_study.txt\", \n",
    "    identifier = \"\", \n",
    "    title = \"Multi-omics data of a Czech population cohort\",\n",
    "    description = \"Multi-omics data of a Czech population cohort\", \n",
    "    submission_date = \"\", \n",
    "    public_release_date = \"\",\n",
    "    contacts = [\n",
    "        Person(\n",
    "            last_name = \"Hajduch\", \n",
    "            first_name = \"Marian\",\n",
    "            #mid_initials = \"\",\n",
    "            affiliation = \"Institute of Molecular and Translational Medicine (IMTM), Palacký University Olomouc\")],\n",
    "    design_descriptors = [\n",
    "        OntologyAnnotation(\n",
    "                term = \"Multi-omics study\",\n",
    "                term_source = ontologies[\"PRIDE\"],\n",
    "                term_accession = \"http://purl.obolibrary.org/obo/PRIDE_0000461\"),\n",
    "        OntologyAnnotation(\n",
    "                term = \"population based study design\",\n",
    "                term_source = ontologies[\"OMIABIS\"],\n",
    "                term_accession = \"http://purl.obolibrary.org/obo/OMIABIS_0001022\")], \n",
    "    factors = None, \n",
    "    protocols = None,\n",
    "    assays = None,\n",
    "    sources = None,\n",
    "    samples = None,\n",
    "    process_sequence = None,\n",
    "    other_material = None,\n",
    "    characteristic_categories = None,\n",
    "    comments = None,\n",
    "    units = None)\n",
    "investigation.studies.append(cohort_study)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Define sample collection protocol"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# sample collection protocol and protocol parameters \n",
    "protocol_params = {\n",
    "    \"anatomical entity\": ProtocolParameter(\n",
    "        parameter_name = OntologyAnnotation(\n",
    "            term = \"anatomical entity\",\n",
    "            term_source = ontologies[\"UBERON\"],\n",
    "            term_accession = \"http://purl.obolibrary.org/obo/UBERON_0001062\"))}\n",
    "\n",
    "protocols = {\n",
    "    \"sample_collection\": Protocol( \n",
    "        name = \"sample_collection_protocol\",\n",
    "        protocol_type = OntologyAnnotation(\n",
    "            term = \"sample collection\",\n",
    "            term_source = None, #ontologies[\"\"], \n",
    "            term_accession = None),\n",
    "        parameters = [protocol_params[\"anatomical entity\"]]),\n",
    "    \"dna_extraction\": Protocol(\n",
    "        name = \"DNA extraction\",\n",
    "        protocol_type = OntologyAnnotation(\n",
    "            term = \"DNA extraction\",\n",
    "            term_source = ontologies[\"OBI\"], \n",
    "            term_accession = \"http://purl.obolibrary.org/obo/OBI_0000257\")),\n",
    "    \"rna_extraction\": Protocol(\n",
    "        name = \"RNA extraction\",\n",
    "        protocol_type = OntologyAnnotation(\n",
    "            term = \"RNA extraction\",\n",
    "            term_source = ontologies[\"OBI\"], \n",
    "            term_accession = \"http://purl.obolibrary.org/obo/OBI_0666666\")),\n",
    "    \"WGS\": Protocol(\n",
    "        name = \"Whole Genome Sequencing\", \n",
    "        protocol_type = OntologyAnnotation(\n",
    "            term = \"Whole Genome Sequencing\",\n",
    "            term_source = ontologies[\"NCIT\"],\n",
    "        term_accession = \"http://purl.obolibrary.org/obo/NCIT_C101294\")),\n",
    "    \"RNAseq\": Protocol(\n",
    "        name = \"mRNA Sequencing\", \n",
    "        protocol_type = OntologyAnnotation(\n",
    "            term = \"mRNA Sequencing\",\n",
    "            term_source = ontologies[\"NCIT\"],\n",
    "        term_accession = \"http://purl.obolibrary.org/obo/NCIT_C129432\")),\n",
    "    \"microRNAseq\": Protocol(\n",
    "        name = \"MicroRNA Sequencing\", \n",
    "        protocol_type = OntologyAnnotation(\n",
    "            term = \"MicroRNA Sequencing\",\n",
    "            term_source = ontologies[\"NCIT\"],\n",
    "        term_accession = \"http://purl.obolibrary.org/obo/NCIT_C156057\"))\n",
    "}\n",
    "for protocol in protocols.values():\n",
    "    cohort_study.protocols.append(protocol)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Define sources and derived samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# add dummy samples\n",
    "for source_idx in range(1, 4):\n",
    "    # create source (=individual)\n",
    "    source_name = \"individual_{0}\".format(source_idx)\n",
    "    source = Source(\n",
    "        name = source_name,\n",
    "        characteristics = [\n",
    "            Characteristic(\n",
    "                category = OntologyAnnotation(\n",
    "                    term = \"Organism\",\n",
    "                    term_source = ontologies[\"OBI\"],\n",
    "                    term_accession = \"http://purl.obolibrary.org/obo/OBI_0100026\"),\n",
    "                value = OntologyAnnotation(\n",
    "                    term = \"Homo sapiens\",\n",
    "                    term_source = ontologies[\"NCBITAXON\"],\n",
    "                    term_accession = \"http://purl.obolibrary.org/obo/NCBITaxon_9606\"))])\n",
    "    cohort_study.sources.append(source)\n",
    "    # create sample\n",
    "    sample_name = \"sample_{0}\".format(source_idx)\n",
    "    sample = Sample(\n",
    "        name = sample_name, \n",
    "        derives_from = source)\n",
    "    cohort_study.samples.append(sample)\n",
    "    # sample collection process\n",
    "    sample_collection_process = Process(\n",
    "        name = \"samplecollection_{0}\".format(source_idx),\n",
    "        executes_protocol = protocols[\"sample_collection\"],\n",
    "        parameter_values = [\n",
    "            ParameterValue(\n",
    "                category = protocol_params[\"anatomical entity\"], #ProtocolParameter \n",
    "                value = OntologyAnnotation(\n",
    "                    term = \"blood\",\n",
    "                    term_source = ontologies[\"UBERON\"],\n",
    "                    term_accession = \"http://purl.obolibrary.org/obo/UBERON_0000178\"))],\n",
    "        inputs = [source],\n",
    "        outputs = [sample])\n",
    "    cohort_study.process_sequence.append(sample_collection_process)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Define assays"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "assays = {\n",
    "    \"genomics_imtm\": Assay(\n",
    "        filename = \"a_assay_genomics_imtm.txt\",\n",
    "        measurement_type = OntologyAnnotation(\n",
    "            term = \"DNA Sequence\",\n",
    "            term_source = ontologies[\"NCIT\"],\n",
    "            term_accession = \"http://purl.obolibrary.org/obo/NCIT_C13299\"),\n",
    "        technology_type = OntologyAnnotation(\n",
    "            term = \"Whole Genome Sequencing\",\n",
    "            term_source = ontologies[\"NCIT\"], \n",
    "            term_accession = \"http://purl.obolibrary.org/obo/NCIT_C101294\"),\n",
    "        technology_platform = OntologyAnnotation(\n",
    "            term = \"Illumina platform\",\n",
    "            term_source = ontologies[\"GENEPIO\"], \n",
    "            term_accession = \"http://purl.obolibrary.org/obo/GENEPIO_0001923\")\n",
    "    ), \n",
    "#    \"dnamethylation_uu\": Assay(\n",
    "#        filename = \"a_assay_dnamethylation_uu.txt\"),\n",
    "#        measurement_type = OntologyAnnotation(\n",
    "#            term = \"Methylation Beta Value\",\n",
    "#            term_source = ontologies[\"NCIT\"],\n",
    "#            term_accession = \"http://purl.obolibrary.org/obo/NCIT_C164051\"),\n",
    "#        technology_type = OntologyAnnotation(\n",
    "#            term = \"DNA methylation profiling by array assay\",\n",
    "#            term_source = ontologies[\"OBI\"], \n",
    "#            term_accession = \"http://purl.obolibrary.org/obo/OBI_0001332\"),\n",
    "#        technology_platform = OntologyAnnotation(\n",
    "#            term = \"Illumina Infinium MethylationEPIC BeadChip\",\n",
    "#            term_source = ontologies[\"OBI\"], \n",
    "#            term_accession = \"http://purl.obolibrary.org/obo/OBI_0002131\")), \n",
    "    \"rnaseq_fimm\": Assay(\n",
    "        # https://eatris.sharepoint.com/:w:/r/sites/eatrisplusgroup/_layouts/15/Doc.aspx?sourcedoc=%7B4966EA29-2AE7-459A-8F58-A3E5962B6988%7D&file=RNAseq%20workflow.docx&action=default&mobileredirect=true\n",
    "        filename = \"a_assay_rnaseq_fimm.txt\",\n",
    "        measurement_type = OntologyAnnotation(\n",
    "            term = \"Binary format\",\n",
    "            term_source = ontologies[\"EDAM\"],\n",
    "            term_accession = \"http://edamontology.org/format_2333\"),\n",
    "        technology_type = OntologyAnnotation(\n",
    "            term = \"mRNA Sequencing\",\n",
    "            term_source = ontologies[\"NCIT\"], \n",
    "            term_accession = \"http://purl.obolibrary.org/obo/NCIT_C129432\"),\n",
    "        technology_platform = OntologyAnnotation(\n",
    "            term = \"Illumina NovaSeq 6000\",\n",
    "            term_source = ontologies[\"OBI\"], \n",
    "            term_accession = \"http://purl.obolibrary.org/obo/OBI_0002630\"))\n",
    "#    \"mirnaseq_fimm\": Assay(\n",
    "#        filename = \"a_assay_mirnaseq_fimm.txt\"), \n",
    "#    \"mirnaseq_sermas\": Assay(\n",
    "#        filename = \"a_assay_mirnaseq_sermas.txt\"), \n",
    "#    \"proteomics_imtm\": Assay(\n",
    "#        filename = \"a_assay_proteomics_imtm.txt\"), \n",
    "#    \"metabolomics_acylcarnitines_rumc\": Assay(\n",
    "#        filename = \"a_assay_metabolomics_acylcarnitines_rumc.txt\"), \n",
    "#    \"metabolomics_aminoacids_mumc\": Assay(\n",
    "#        filename = \"a_assay_metabolomics_aminoacids_mumc.txt\"), \n",
    "#    \"metabolomics_fattyacids_mumc\": Assay(\n",
    "#        filename = \"a_assay_metabolomics_fattyacids_mumc.txt\"), \n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Data file types are:\n",
    "'Raw Data File',\n",
    "'Derived Data File',\n",
    "'Image File',\n",
    "'Acquisition Parameter Data File',\n",
    "'Derived Spectral Data File',\n",
    "'Protein Assignment File',\n",
    "'Raw Spectral Data File',\n",
    "'Peptide Assignment File',\n",
    "'Array Data File',\n",
    "'Derived Array Data File',\n",
    "'Post Translational Modification Assignment File',\n",
    "'Derived Array Data Matrix File',\n",
    "'Free Induction Decay Data File',\n",
    "'Metabolite Assignment File',\n",
    "'Array Data Matrix File'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "for idx, sample in enumerate(cohort_study.samples):\n",
    "    # DNA extraction\n",
    "    dna = Material(\n",
    "        name = \"DNA_{0}\".format(sample.name),\n",
    "        type_ = \"Extract Name\")\n",
    "    dna_extraction_process = Process(\n",
    "        name = \"DNA_extraction_{0}\".format(sample.name),\n",
    "        executes_protocol = protocols[\"dna_extraction\"],\n",
    "        inputs = [sample], \n",
    "        outputs = [dna])\n",
    "    # RNA extraction\n",
    "    rna = Material(\n",
    "        name = \"RNA_{0}\".format(sample.name),\n",
    "        type_ = \"Extract Name\")\n",
    "    rna_extraction_process = Process(\n",
    "        name = \"RNA_extraction_{0}\".format(sample.name),\n",
    "        executes_protocol = protocols[\"rna_extraction\"],\n",
    "        inputs = [sample], \n",
    "        outputs = [rna])\n",
    "    # genomics measurement\n",
    "    wgs_raw_file = DataFile(\n",
    "        filename = \"WGS_rawdata_{0}\".format(dna.name), \n",
    "        label = \"Raw Data File\", \n",
    "        generated_from = [dna])\n",
    "    wgs_process = Process(\n",
    "        name = \"WGS_{0}\".format(dna.name),\n",
    "        executes_protocol = protocols[\"WGS\"],\n",
    "        inputs = [dna], \n",
    "        outputs = [wgs_raw_file])\n",
    "    plink(dna_extraction_process, wgs_process)\n",
    "    assays[\"genomics_imtm\"].samples.append(sample)\n",
    "    assays[\"genomics_imtm\"].data_files.append(wgs_raw_file)\n",
    "    assays[\"genomics_imtm\"].other_material.append(dna)\n",
    "    assays[\"genomics_imtm\"].process_sequence.append(dna_extraction_process)\n",
    "    assays[\"genomics_imtm\"].process_sequence.append(wgs_process)\n",
    "    # transcriptomics - RNAseq\n",
    "    rnaseq_binarybasecall_file = DataFile(\n",
    "        filename = \"RNAseq_BCL_{0}\".format(rna.name), \n",
    "        label = \"Raw Data File\", \n",
    "        generated_from = [rna])\n",
    "    rnaseq_process = Process(\n",
    "        name = \"RNAseq_{0}\".format(rna.name),\n",
    "        executes_protocol = protocols[\"RNAseq\"],\n",
    "        inputs = [rna], \n",
    "        outputs = [rnaseq_binarybasecall_file])\n",
    "    plink(rna_extraction_process, rnaseq_process)\n",
    "    assays[\"rnaseq_fimm\"].samples.append(sample)\n",
    "    assays[\"rnaseq_fimm\"].data_files.append(rnaseq_binarybasecall_file)\n",
    "    assays[\"rnaseq_fimm\"].other_material.append(rna)\n",
    "    assays[\"rnaseq_fimm\"].process_sequence.append(rna_extraction_process)\n",
    "    assays[\"rnaseq_fimm\"].process_sequence.append(rnaseq_process)\n",
    "    # microRNAseq - FIMM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# add assays to study\n",
    "for assay in assays.values():\n",
    "    cohort_study.assays.append(assay)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Write ISA-Tab files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "isatools.model.Investigation(identifier='', filename='i_investigation.txt', title='EATRIS-Plus - Flagship in Personalised Medicine', submission_date='', public_release_date='', ontology_source_references=[isatools.model.OntologySource(name='CHEBI - Chemical Entities of Biological Interest', file='http://purl.obolibrary.org/obo/chebi.owl', version='', description='A structured classification of molecular entities of biological interest focusing on 'small' chemical compounds.', comments=[]), isatools.model.OntologySource(name='CRO - Contributor Role Ontology', file='http://purl.obolibrary.org/obo/cro.owl', version='', description='A classification of the diverse roles performed in the work leading to a published research output in the sciences. Its purpose to provide transparency in contributions to scholarly published work, to enable improved systems of attribution, credit, and accountability.', comments=[]), isatools.model.OntologySource(name='EDAM - EMBRACE Data and Methods', file='http://edamontology.org/EDAM.owl', version='', description='EDAM (EMBRACE Data and Methods) is an ontology of common bioinformatics operations, topics, types of data including identifiers, and formats. EDAM comprises common concepts (shared within the bioinformatics community) that apply to semantic annotation of resources.', comments=[]), isatools.model.OntologySource(name='EFO - Experimental Factor Ontology', file='http://www.ebi.ac.uk/efo/efo.owl', version='', description='The Experimental Factor Ontology (EFO) provides a systematic description of many experimental variables available in EBI databases, and for external projects such as the NHGRI GWAS catalogue. It combines parts of several biological ontologies, such as anatomy, disease and chemical compounds. The scope of EFO is to support the annotation, analysis and visualization of data handled by many groups at the EBI and as the core ontology for OpenTargets.org', comments=[]), isatools.model.OntologySource(name='GENEPIO - Genomic Epidemiology Ontology', file='http://purl.obolibrary.org/obo/genepio.owl', version='', description='The Genomic Epidemiology Ontology (GenEpiO) covers vocabulary necessary to identify, document and research foodborne pathogens and associated outbreaks.', comments=[]), isatools.model.OntologySource(name='MMO - Measurement method ontology', file='http://purl.obolibrary.org/obo/mmo.owl', version='', description='A representation of the variety of methods used to make clinical and phenotype measurements.', comments=[]), isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), isatools.model.OntologySource(name='NCI Thesaurus OBO Edition', file='http://purl.obolibrary.org/obo/ncit.owl', version='', description='The NCIt OBO Edition project aims to increase integration of the NCIt with OBO Library ontologies. NCIt is a reference terminology that includes broad coverage of the cancer domain, including cancer related diseases, findings and abnormalities. NCIt OBO Edition releases should be considered experimental.', comments=[]), isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), isatools.model.OntologySource(name='Ontologized MIABIS', file='http://purl.obolibrary.org/obo/omiabis.owl', version='', description='An ontological version of MIABIS (Minimum Information About BIobank data Sharing)', comments=[]), isatools.model.OntologySource(name='PRIDE Controlled Vocabulary', file='http://purl.obolibrary.org/obo/pride_cv.obo', version='', description='The PRIDE PRoteomics IDEntifications (PRIDE) database is a centralized, standards compliant, public data repository for proteomics data, including protein and peptide identifications, post-translational modifications and supporting spectral evidence.', comments=[]), isatools.model.OntologySource(name='STATO: the statistical methods ontology', file='http://purl.obolibrary.org/obo/stato.owl', version='', description='STATO is the statistical methods ontology. It contains concepts and properties related to statistical methods, probability distributions and other concepts related to statistical analysis, including relationships to study designs and plots.', comments=[]), isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[])], publications=[], contacts=[isatools.model.Person(last_name='Keidong', first_name='Eliis', mid_initials='', email='', phone='', fax='', address='', affiliation='EATRIS', roles=[isatools.model.OntologyAnnotation(term='project management role', term_source=isatools.model.OntologySource(name='CRO - Contributor Role Ontology', file='http://purl.obolibrary.org/obo/cro.owl', version='', description='A classification of the diverse roles performed in the work leading to a published research output in the sciences. Its purpose to provide transparency in contributions to scholarly published work, to enable improved systems of attribution, credit, and accountability.', comments=[]), term_accession='http://purl.obolibrary.org/obo/CRO_0000065', comments=[])], comments=[])], studies=[isatools.model.Study(filename='s_study.txt', identifier='', title='Multi-omics data of a Czech population cohort', description='Multi-omics data of a Czech population cohort', submission_date='', public_release_date='', contacts=[isatools.model.Person(last_name='Hajduch', first_name='Marian', mid_initials='', email='', phone='', fax='', address='', affiliation='Institute of Molecular and Translational Medicine (IMTM), Palacký University Olomouc', roles=[], comments=[])], design_descriptors=[isatools.model.OntologyAnnotation(term='Multi-omics study', term_source=isatools.model.OntologySource(name='PRIDE Controlled Vocabulary', file='http://purl.obolibrary.org/obo/pride_cv.obo', version='', description='The PRIDE PRoteomics IDEntifications (PRIDE) database is a centralized, standards compliant, public data repository for proteomics data, including protein and peptide identifications, post-translational modifications and supporting spectral evidence.', comments=[]), term_accession='http://purl.obolibrary.org/obo/PRIDE_0000461', comments=[]), isatools.model.OntologyAnnotation(term='population based study design', term_source=isatools.model.OntologySource(name='Ontologized MIABIS', file='http://purl.obolibrary.org/obo/omiabis.owl', version='', description='An ontological version of MIABIS (Minimum Information About BIobank data Sharing)', comments=[]), term_accession='http://purl.obolibrary.org/obo/OMIABIS_0001022', comments=[])], publications=[], factors=[], protocols=[isatools.model.Protocol(name='sample_collection_protocol', protocol_type=isatools.model.OntologyAnnotation(term='sample collection', term_source=None, term_accession='None', comments=[]), uri='', version='', parameters=[isatools.model.ProtocolParameter(parameter_name=isatools.model.OntologyAnnotation(term='anatomical entity', term_source=isatools.model.OntologySource(name='Uber-anatomy ontology', file='http://purl.obolibrary.org/obo/uberon.owl', version='', description='Uberon is an integrated cross-species anatomy ontology representing a variety of entities classified according to traditional anatomical criteria such as structure, function and developmental lineage. The ontology includes comprehensive relationships to taxon-specific anatomical ontologies, allowing integration of functional, phenotype and expression data.', comments=[]), term_accession='http://purl.obolibrary.org/obo/UBERON_0001062', comments=[]), comments=[])], components=[], comments=[]), isatools.model.Protocol(name='DNA extraction', protocol_type=isatools.model.OntologyAnnotation(term='DNA extraction', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0000257', comments=[]), uri='', version='', parameters=[], components=[], comments=[]), isatools.model.Protocol(name='RNA extraction', protocol_type=isatools.model.OntologyAnnotation(term='RNA extraction', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0666666', comments=[]), uri='', version='', parameters=[], components=[], comments=[]), isatools.model.Protocol(name='Whole Genome Sequencing', protocol_type=isatools.model.OntologyAnnotation(term='Whole Genome Sequencing', term_source=isatools.model.OntologySource(name='NCI Thesaurus OBO Edition', file='http://purl.obolibrary.org/obo/ncit.owl', version='', description='The NCIt OBO Edition project aims to increase integration of the NCIt with OBO Library ontologies. NCIt is a reference terminology that includes broad coverage of the cancer domain, including cancer related diseases, findings and abnormalities. NCIt OBO Edition releases should be considered experimental.', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCIT_C101294', comments=[]), uri='', version='', parameters=[], components=[], comments=[]), isatools.model.Protocol(name='mRNA Sequencing', protocol_type=isatools.model.OntologyAnnotation(term='mRNA Sequencing', term_source=isatools.model.OntologySource(name='NCI Thesaurus OBO Edition', file='http://purl.obolibrary.org/obo/ncit.owl', version='', description='The NCIt OBO Edition project aims to increase integration of the NCIt with OBO Library ontologies. NCIt is a reference terminology that includes broad coverage of the cancer domain, including cancer related diseases, findings and abnormalities. NCIt OBO Edition releases should be considered experimental.', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCIT_C129432', comments=[]), uri='', version='', parameters=[], components=[], comments=[]), isatools.model.Protocol(name='MicroRNA Sequencing', protocol_type=isatools.model.OntologyAnnotation(term='MicroRNA Sequencing', term_source=isatools.model.OntologySource(name='NCI Thesaurus OBO Edition', file='http://purl.obolibrary.org/obo/ncit.owl', version='', description='The NCIt OBO Edition project aims to increase integration of the NCIt with OBO Library ontologies. NCIt is a reference terminology that includes broad coverage of the cancer domain, including cancer related diseases, findings and abnormalities. NCIt OBO Edition releases should be considered experimental.', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCIT_C156057', comments=[]), uri='', version='', parameters=[], components=[], comments=[])], assays=[isatools.model.Assay(measurement_type=isatools.model.OntologyAnnotation(term='DNA Sequence', term_source=isatools.model.OntologySource(name='NCI Thesaurus OBO Edition', file='http://purl.obolibrary.org/obo/ncit.owl', version='', description='The NCIt OBO Edition project aims to increase integration of the NCIt with OBO Library ontologies. NCIt is a reference terminology that includes broad coverage of the cancer domain, including cancer related diseases, findings and abnormalities. NCIt OBO Edition releases should be considered experimental.', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCIT_C13299', comments=[]), technology_type=isatools.model.OntologyAnnotation(term='Whole Genome Sequencing', term_source=isatools.model.OntologySource(name='NCI Thesaurus OBO Edition', file='http://purl.obolibrary.org/obo/ncit.owl', version='', description='The NCIt OBO Edition project aims to increase integration of the NCIt with OBO Library ontologies. NCIt is a reference terminology that includes broad coverage of the cancer domain, including cancer related diseases, findings and abnormalities. NCIt OBO Edition releases should be considered experimental.', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCIT_C101294', comments=[]), technology_platform='OntologyAnnotation(\n",
       "    term=Illumina platform\n",
       "    term_source=GENEPIO - Genomic Epidemiology Ontology\n",
       "    term_accession=http://purl.obolibrary.org/obo/GENEPIO_0001923\n",
       "    comments=0 Comment objects\n",
       ")', filename='a_assay_genomics_imtm.txt', data_files=[isatools.model.DataFile(filename='WGS_rawdata_DNA_sample_1', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7ff67a86c2b0>], comments=[]), isatools.model.DataFile(filename='WGS_rawdata_DNA_sample_2', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7ff67a86c7c0>], comments=[]), isatools.model.DataFile(filename='WGS_rawdata_DNA_sample_3', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7ff67a86cac0>], comments=[])], samples=[isatools.model.Sample(name='sample_1', characteristics=[], factor_values=[], derives_from=Source(\n",
       "    name=individual_1\n",
       "    characteristics=1 Characteristic objects\n",
       "    comments=0 Comment objects\n",
       "), comments=[]), isatools.model.Sample(name='sample_2', characteristics=[], factor_values=[], derives_from=Source(\n",
       "    name=individual_2\n",
       "    characteristics=1 Characteristic objects\n",
       "    comments=0 Comment objects\n",
       "), comments=[]), isatools.model.Sample(name='sample_3', characteristics=[], factor_values=[], derives_from=Source(\n",
       "    name=individual_3\n",
       "    characteristics=1 Characteristic objects\n",
       "    comments=0 Comment objects\n",
       "), comments=[])], process_sequence=[isatools.model.Process(id=\"\". name=\"DNA_extraction_sample_1\", executes_protocol=Protocol(\n",
       "    name=DNA extraction\n",
       "    protocol_type=DNA extraction\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Sample(name='sample_1', characteristics=[], factor_values=[], derives_from=Source(\n",
       "    name=individual_1\n",
       "    characteristics=1 Characteristic objects\n",
       "    comments=0 Comment objects\n",
       "), comments=[])], outputs=[<isatools.model.Material object at 0x7ff67a86c2b0>]), isatools.model.Process(id=\"\". name=\"WGS_DNA_sample_1\", executes_protocol=Protocol(\n",
       "    name=Whole Genome Sequencing\n",
       "    protocol_type=Whole Genome Sequencing\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7ff67a86c2b0>], outputs=[isatools.model.DataFile(filename='WGS_rawdata_DNA_sample_1', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7ff67a86c2b0>], comments=[])]), isatools.model.Process(id=\"\". name=\"DNA_extraction_sample_2\", executes_protocol=Protocol(\n",
       "    name=DNA extraction\n",
       "    protocol_type=DNA extraction\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Sample(name='sample_2', characteristics=[], factor_values=[], derives_from=Source(\n",
       "    name=individual_2\n",
       "    characteristics=1 Characteristic objects\n",
       "    comments=0 Comment objects\n",
       "), comments=[])], outputs=[<isatools.model.Material object at 0x7ff67a86c7c0>]), isatools.model.Process(id=\"\". name=\"WGS_DNA_sample_2\", executes_protocol=Protocol(\n",
       "    name=Whole Genome Sequencing\n",
       "    protocol_type=Whole Genome Sequencing\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7ff67a86c7c0>], outputs=[isatools.model.DataFile(filename='WGS_rawdata_DNA_sample_2', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7ff67a86c7c0>], comments=[])]), isatools.model.Process(id=\"\". name=\"DNA_extraction_sample_3\", executes_protocol=Protocol(\n",
       "    name=DNA extraction\n",
       "    protocol_type=DNA extraction\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Sample(name='sample_3', characteristics=[], factor_values=[], derives_from=Source(\n",
       "    name=individual_3\n",
       "    characteristics=1 Characteristic objects\n",
       "    comments=0 Comment objects\n",
       "), comments=[])], outputs=[<isatools.model.Material object at 0x7ff67a86cac0>]), isatools.model.Process(id=\"\". name=\"WGS_DNA_sample_3\", executes_protocol=Protocol(\n",
       "    name=Whole Genome Sequencing\n",
       "    protocol_type=Whole Genome Sequencing\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7ff67a86cac0>], outputs=[isatools.model.DataFile(filename='WGS_rawdata_DNA_sample_3', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7ff67a86cac0>], comments=[])])], other_material=[<isatools.model.Material object at 0x7ff67a86c2b0>, <isatools.model.Material object at 0x7ff67a86c7c0>, <isatools.model.Material object at 0x7ff67a86cac0>], characteristic_categories=[], comments=[], units=[]), isatools.model.Assay(measurement_type=isatools.model.OntologyAnnotation(term='Binary format', term_source=isatools.model.OntologySource(name='EDAM - EMBRACE Data and Methods', file='http://edamontology.org/EDAM.owl', version='', description='EDAM (EMBRACE Data and Methods) is an ontology of common bioinformatics operations, topics, types of data including identifiers, and formats. EDAM comprises common concepts (shared within the bioinformatics community) that apply to semantic annotation of resources.', comments=[]), term_accession='http://edamontology.org/format_2333', comments=[]), technology_type=isatools.model.OntologyAnnotation(term='mRNA Sequencing', term_source=isatools.model.OntologySource(name='NCI Thesaurus OBO Edition', file='http://purl.obolibrary.org/obo/ncit.owl', version='', description='The NCIt OBO Edition project aims to increase integration of the NCIt with OBO Library ontologies. NCIt is a reference terminology that includes broad coverage of the cancer domain, including cancer related diseases, findings and abnormalities. NCIt OBO Edition releases should be considered experimental.', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCIT_C129432', comments=[]), technology_platform='OntologyAnnotation(\n",
       "    term=Illumina NovaSeq 6000\n",
       "    term_source=OBI - Ontology for Biomedical Investigations\n",
       "    term_accession=http://purl.obolibrary.org/obo/OBI_0002630\n",
       "    comments=0 Comment objects\n",
       ")', filename='a_assay_rnaseq_fimm.txt', data_files=[isatools.model.DataFile(filename='RNAseq_BCL_RNA_sample_1', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7ff67a86c640>], comments=[]), isatools.model.DataFile(filename='RNAseq_BCL_RNA_sample_2', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7ff67a86c880>], comments=[]), isatools.model.DataFile(filename='RNAseq_BCL_RNA_sample_3', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7ff67a86cb50>], comments=[])], samples=[isatools.model.Sample(name='sample_1', characteristics=[], factor_values=[], derives_from=Source(\n",
       "    name=individual_1\n",
       "    characteristics=1 Characteristic objects\n",
       "    comments=0 Comment objects\n",
       "), comments=[]), isatools.model.Sample(name='sample_2', characteristics=[], factor_values=[], derives_from=Source(\n",
       "    name=individual_2\n",
       "    characteristics=1 Characteristic objects\n",
       "    comments=0 Comment objects\n",
       "), comments=[]), isatools.model.Sample(name='sample_3', characteristics=[], factor_values=[], derives_from=Source(\n",
       "    name=individual_3\n",
       "    characteristics=1 Characteristic objects\n",
       "    comments=0 Comment objects\n",
       "), comments=[])], process_sequence=[isatools.model.Process(id=\"\". name=\"RNA_extraction_sample_1\", executes_protocol=Protocol(\n",
       "    name=RNA extraction\n",
       "    protocol_type=RNA extraction\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Sample(name='sample_1', characteristics=[], factor_values=[], derives_from=Source(\n",
       "    name=individual_1\n",
       "    characteristics=1 Characteristic objects\n",
       "    comments=0 Comment objects\n",
       "), comments=[])], outputs=[<isatools.model.Material object at 0x7ff67a86c640>]), isatools.model.Process(id=\"\". name=\"RNAseq_RNA_sample_1\", executes_protocol=Protocol(\n",
       "    name=mRNA Sequencing\n",
       "    protocol_type=mRNA Sequencing\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7ff67a86c640>], outputs=[isatools.model.DataFile(filename='RNAseq_BCL_RNA_sample_1', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7ff67a86c640>], comments=[])]), isatools.model.Process(id=\"\". name=\"RNA_extraction_sample_2\", executes_protocol=Protocol(\n",
       "    name=RNA extraction\n",
       "    protocol_type=RNA extraction\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Sample(name='sample_2', characteristics=[], factor_values=[], derives_from=Source(\n",
       "    name=individual_2\n",
       "    characteristics=1 Characteristic objects\n",
       "    comments=0 Comment objects\n",
       "), comments=[])], outputs=[<isatools.model.Material object at 0x7ff67a86c880>]), isatools.model.Process(id=\"\". name=\"RNAseq_RNA_sample_2\", executes_protocol=Protocol(\n",
       "    name=mRNA Sequencing\n",
       "    protocol_type=mRNA Sequencing\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7ff67a86c880>], outputs=[isatools.model.DataFile(filename='RNAseq_BCL_RNA_sample_2', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7ff67a86c880>], comments=[])]), isatools.model.Process(id=\"\". name=\"RNA_extraction_sample_3\", executes_protocol=Protocol(\n",
       "    name=RNA extraction\n",
       "    protocol_type=RNA extraction\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Sample(name='sample_3', characteristics=[], factor_values=[], derives_from=Source(\n",
       "    name=individual_3\n",
       "    characteristics=1 Characteristic objects\n",
       "    comments=0 Comment objects\n",
       "), comments=[])], outputs=[<isatools.model.Material object at 0x7ff67a86cb50>]), isatools.model.Process(id=\"\". name=\"RNAseq_RNA_sample_3\", executes_protocol=Protocol(\n",
       "    name=mRNA Sequencing\n",
       "    protocol_type=mRNA Sequencing\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=0 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[<isatools.model.Material object at 0x7ff67a86cb50>], outputs=[isatools.model.DataFile(filename='RNAseq_BCL_RNA_sample_3', label='Raw Data File', generated_from=[<isatools.model.Material object at 0x7ff67a86cb50>], comments=[])])], other_material=[<isatools.model.Material object at 0x7ff67a86c640>, <isatools.model.Material object at 0x7ff67a86c880>, <isatools.model.Material object at 0x7ff67a86cb50>], characteristic_categories=[], comments=[], units=[])], sources=[isatools.model.Source(name='individual_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[]), isatools.model.Source(name='individual_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[]), isatools.model.Source(name='individual_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], samples=[isatools.model.Sample(name='sample_1', characteristics=[], factor_values=[], derives_from=Source(\n",
       "    name=individual_1\n",
       "    characteristics=1 Characteristic objects\n",
       "    comments=0 Comment objects\n",
       "), comments=[]), isatools.model.Sample(name='sample_2', characteristics=[], factor_values=[], derives_from=Source(\n",
       "    name=individual_2\n",
       "    characteristics=1 Characteristic objects\n",
       "    comments=0 Comment objects\n",
       "), comments=[]), isatools.model.Sample(name='sample_3', characteristics=[], factor_values=[], derives_from=Source(\n",
       "    name=individual_3\n",
       "    characteristics=1 Characteristic objects\n",
       "    comments=0 Comment objects\n",
       "), comments=[])], process_sequence=[isatools.model.Process(id=\"\". name=\"samplecollection_1\", executes_protocol=Protocol(\n",
       "    name=sample_collection_protocol\n",
       "    protocol_type=sample collection\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=1 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Source(name='individual_1', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], outputs=[isatools.model.Sample(name='sample_1', characteristics=[], factor_values=[], derives_from=Source(\n",
       "    name=individual_1\n",
       "    characteristics=1 Characteristic objects\n",
       "    comments=0 Comment objects\n",
       "), comments=[])]), isatools.model.Process(id=\"\". name=\"samplecollection_2\", executes_protocol=Protocol(\n",
       "    name=sample_collection_protocol\n",
       "    protocol_type=sample collection\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=1 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Source(name='individual_2', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], outputs=[isatools.model.Sample(name='sample_2', characteristics=[], factor_values=[], derives_from=Source(\n",
       "    name=individual_2\n",
       "    characteristics=1 Characteristic objects\n",
       "    comments=0 Comment objects\n",
       "), comments=[])]), isatools.model.Process(id=\"\". name=\"samplecollection_3\", executes_protocol=Protocol(\n",
       "    name=sample_collection_protocol\n",
       "    protocol_type=sample collection\n",
       "    uri=\n",
       "    version=\n",
       "    parameters=1 ProtocolParameter objects\n",
       "    components=0 OntologyAnnotation objects\n",
       "    comments=0 Comment objects\n",
       "), date=\"None\", performer=\"None\", inputs=[isatools.model.Source(name='individual_3', characteristics=[isatools.model.Characteristic(category=isatools.model.OntologyAnnotation(term='Organism', term_source=isatools.model.OntologySource(name='OBI - Ontology for Biomedical Investigations', file='http://purl.obolibrary.org/obo/obi.owl', version='', description='An integrated ontology for the description of life-science and clinical investigations', comments=[]), term_accession='http://purl.obolibrary.org/obo/OBI_0100026', comments=[]), value=isatools.model.OntologyAnnotation(term='Homo sapiens', term_source=isatools.model.OntologySource(name='NCBI organismal classification', file='http://purl.obolibrary.org/obo/ncbitaxon.owl', version='', description='An ontology representation of the NCBI organismal taxonomy', comments=[]), term_accession='http://purl.obolibrary.org/obo/NCBITaxon_9606', comments=[]), unit=None, comments=[])], comments=[])], outputs=[isatools.model.Sample(name='sample_3', characteristics=[], factor_values=[], derives_from=Source(\n",
       "    name=individual_3\n",
       "    characteristics=1 Characteristic objects\n",
       "    comments=0 Comment objects\n",
       "), comments=[])])], other_material=[], characteristic_categories=[], comments=[], units=[])], comments=[])"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# write to ISA-Tab\n",
    "import os\n",
    "out_dir = \".\"\n",
    "if not os.path.isdir(out_dir):\n",
    "    os.makedirs(out_dir)\n",
    "from isatools import isatab\n",
    "isatab.dump(investigation, out_dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# write to ISA-JSON\n",
    "# see example: https://isa-tools.org/isa-api/content/examples/example-createSimpleISAJSON.html\n",
    "import json\n",
    "from isatools.isajson import ISAJSONEncoder\n",
    "with open(os.path.join(out_dir, \"isa.json\"), \"w\") as out_file:\n",
    "    json.dump(\n",
    "        investigation, \n",
    "        out_file,\n",
    "        cls = ISAJSONEncoder, \n",
    "        sort_keys = True, \n",
    "        indent = 4, \n",
    "        separators = (',', ': '))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#from isatools.convert import json2isatab\n",
    "#from isatools import isajson\n",
    "#isajson.validate(open('isa.json'))\n",
    "#with open(\"isa.json\") as file_pointer:\n",
    "#    json2isatab.convert(file_pointer, './ISA/')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:isa_env] *",
   "language": "python",
   "name": "conda-env-isa_env-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
